ccl/kvccl/kvtenantccl: TestTenantUpgradeFailure failed

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3915298&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3915298&tab=artifacts#/) on master @ [acf40e97f64a97338cbebd20d7fbe176c4e37c34](https://github.com/cockroachdb/cockroach/commits/acf40e97f64a97338cbebd20d7fbe176c4e37c34):


```
=== RUN   TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3935587&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3935587&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
I211217 09:10:55.339773 18002 2@util/log/event_log.go:32  [n1] 1685 ={"Timestamp":1639732255339758392,"EventType":"runtime_stats","MemRSSBytes":1331490816,"GoroutineCount":489,"MemStackSysBytes":6062080,"GoAllocBytes":131331720,"GoTotalBytes":213892136,"GoStatsStaleness":0.002559745,"HeapFragmentBytes":26790264,"HeapReservedBytes":33071104,"HeapReleasedBytes":4071424,"CGoAllocBytes":5919712,"CGoTotalBytes":15507456,"CGoCallRate":0.29997146,"CPUUserPercent":69.89335,"CPUSysPercent":5.399486,"GCPausePercent":0.0069991737,"GCRunCount":126,"NetHostRecvBytes":105024,"NetHostSendBytes":105024}
I211217 09:10:55.341223 18002 2@server/status/runtime_log.go:47  [n1] 1686  runtime stats: 1.2 GiB RSS, 489 goroutines (stacks: 5.8 MiB), 125 MiB/204 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 32 MiB, heap released: 3.9 MiB), 5.6 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 69.9/5.4 %(u/s)time, 0.0 %gc (126x), 103 KiB/103 KiB (r/w)net
I211217 09:10:56.628995 17909 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1687  [JOB 6] WAL created 000007
I211217 09:11:04.706679 23617 2@util/log/event_log.go:32  [nsql2] 1688 ={"Timestamp":1639732264706660609,"EventType":"runtime_stats","MemRSSBytes":1374375936,"GoroutineCount":489,"MemStackSysBytes":5013504,"GoAllocBytes":91517760,"GoTotalBytes":221554744,"GoStatsStaleness":0.002636843,"HeapFragmentBytes":31354048,"HeapReservedBytes":74866688,"HeapReleasedBytes":2768896,"CGoAllocBytes":10114144,"CGoTotalBytes":19705856,"CGoCallRate":0.3999428,"CPUUserPercent":69.49007,"CPUSysPercent":5.199256,"GCPausePercent":0.0040039076,"GCRunCount":129,"NetHostRecvBytes":92449,"NetHostSendBytes":92449}
I211217 09:11:04.708296 23617 2@server/status/runtime_log.go:47  [nsql2] 1689  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 4.8 MiB), 87 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 30 MiB, heap reserved: 71 MiB, heap released: 2.6 MiB), 9.6 MiB/19 MiB CGO alloc/total (0.4 CGO/sec), 69.5/5.2 %(u/s)time, 0.0 %gc (129x), 90 KiB/90 KiB (r/w)net
I211217 09:11:05.194650 23868 2@util/log/event_log.go:32  [nsql3] 1690 ={"Timestamp":1639732265194634214,"EventType":"runtime_stats","MemRSSBytes":1366294528,"GoroutineCount":489,"MemStackSysBytes":5308416,"GoAllocBytes":104117128,"GoTotalBytes":221554744,"GoStatsStaleness":0.001955782,"HeapFragmentBytes":29568120,"HeapReservedBytes":63758336,"HeapReleasedBytes":2768896,"CGoAllocBytes":10114144,"CGoTotalBytes":19705856,"CGoCallRate":0.39996842,"CPUUserPercent":69.194534,"CPUSysPercent":4.999605,"GCPausePercent":0.004004164,"GCRunCount":129,"NetHostRecvBytes":91511,"NetHostSendBytes":91511}
I211217 09:11:05.195738 23868 2@server/status/runtime_log.go:47  [nsql3] 1691  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.1 MiB), 99 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 28 MiB, heap reserved: 61 MiB, heap released: 2.6 MiB), 9.6 MiB/19 MiB CGO alloc/total (0.4 CGO/sec), 69.2/5.0 %(u/s)time, 0.0 %gc (129x), 89 KiB/89 KiB (r/w)net
I211217 09:11:05.332571 17979 2@gossip/gossip.go:566  [n1] 1692  gossip status (ok, 1 node)
I211217 09:11:05.332571 17979 2@gossip/gossip.go:566  [n1] 1692 +gossip client (0/3 cur/max conns)
I211217 09:11:05.332571 17979 2@gossip/gossip.go:566  [n1] 1692 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211217 09:11:05.332571 17979 2@gossip/gossip.go:566  [n1] 1692 +gossip connectivity
I211217 09:11:05.332571 17979 2@gossip/gossip.go:566  [n1] 1692 +  n1 [sentinel];
I211217 09:11:05.339206 18002 2@util/log/event_log.go:32  [n1] 1693 ={"Timestamp":1639732265339194694,"EventType":"runtime_stats","MemRSSBytes":1366802432,"GoroutineCount":489,"MemStackSysBytes":5439488,"GoAllocBytes":108375408,"GoTotalBytes":221554744,"GoStatsStaleness":0.001841816,"HeapFragmentBytes":28996240,"HeapReservedBytes":59940864,"HeapReleasedBytes":2768896,"CGoAllocBytes":10114144,"CGoTotalBytes":19705856,"CGoCallRate":0.40002242,"CPUUserPercent":70.403946,"CPUSysPercent":5.2002916,"GCPausePercent":0.0040047048,"GCRunCount":129,"NetHostRecvBytes":92227,"NetHostSendBytes":92227}
I211217 09:11:05.340235 18002 2@server/status/runtime_log.go:47  [n1] 1694  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.2 MiB), 103 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 28 MiB, heap reserved: 57 MiB, heap released: 2.6 MiB), 9.6 MiB/19 MiB CGO alloc/total (0.4 CGO/sec), 70.4/5.2 %(u/s)time, 0.0 %gc (129x), 90 KiB/90 KiB (r/w)net
I211217 09:11:07.049296 18894 server/auto_upgrade.go:44  [n1] 1695  auto upgrade disabled by testing
I211217 09:11:14.706218 23617 2@util/log/event_log.go:32  [nsql2] 1696 ={"Timestamp":1639732274706203999,"EventType":"runtime_stats","MemRSSBytes":1377198080,"GoroutineCount":488,"MemStackSysBytes":5701632,"GoAllocBytes":136194056,"GoTotalBytes":221554744,"GoStatsStaleness":0.001823424,"HeapFragmentBytes":27432952,"HeapReservedBytes":33423360,"HeapReleasedBytes":2768896,"CGoAllocBytes":10114144,"CGoTotalBytes":19705856,"CGoCallRate":0.3000136,"CPUUserPercent":65.10294,"CPUSysPercent":5.2002354,"GCPausePercent":0.0040233126,"GCRunCount":131,"NetHostRecvBytes":92761,"NetHostSendBytes":92761}
I211217 09:11:14.707292 23617 2@server/status/runtime_log.go:47  [nsql2] 1697  runtime stats: 1.3 GiB RSS, 488 goroutines (stacks: 5.4 MiB), 130 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 32 MiB, heap released: 2.6 MiB), 9.6 MiB/19 MiB CGO alloc/total (0.3 CGO/sec), 65.1/5.2 %(u/s)time, 0.0 %gc (131x), 91 KiB/91 KiB (r/w)net
I211217 09:11:14.909920 25575 jobs/registry.go:1179  [nsql2] 1698  AUTO CREATE STATS job 719788031351455746: stepping through state running with error: <nil>
I211217 09:11:15.061158 25575 jobs/registry.go:1179  [nsql2] 1699  AUTO CREATE STATS job 719788031351455746: stepping through state succeeded with error: <nil>
I211217 09:11:15.195211 23868 2@util/log/event_log.go:32  [nsql3] 1700 ={"Timestamp":1639732275195195170,"EventType":"runtime_stats","MemRSSBytes":1331896320,"GoroutineCount":487,"MemStackSysBytes":5210112,"GoAllocBytes":96126240,"GoTotalBytes":222119992,"GoStatsStaleness":0.001927755,"HeapFragmentBytes":29457120,"HeapReservedBytes":72261632,"HeapReleasedBytes":2465792,"CGoAllocBytes":10114144,"CGoTotalBytes":19705856,"CGoCallRate":0.29998305,"CPUUserPercent":71.89594,"CPUSysPercent":6.399638,"GCPausePercent":0.0065051825,"GCRunCount":132,"NetHostRecvBytes":164274,"NetHostSendBytes":164274}
I211217 09:11:15.196246 23868 2@server/status/runtime_log.go:47  [nsql3] 1701  runtime stats: 1.2 GiB RSS, 487 goroutines (stacks: 5.0 MiB), 92 MiB/212 MiB Go alloc/total(stale) (heap fragmentation: 28 MiB, heap reserved: 69 MiB, heap released: 2.4 MiB), 9.6 MiB/19 MiB CGO alloc/total (0.3 CGO/sec), 71.9/6.4 %(u/s)time, 0.0 %gc (132x), 160 KiB/160 KiB (r/w)net
I211217 09:11:15.339839 18002 2@util/log/event_log.go:32  [n1] 1702 ={"Timestamp":1639732275339827442,"EventType":"runtime_stats","MemRSSBytes":1321463808,"GoroutineCount":486,"MemStackSysBytes":5439488,"GoAllocBytes":101409032,"GoTotalBytes":222119992,"GoStatsStaleness":0.001876726,"HeapFragmentBytes":28540664,"HeapReservedBytes":67665920,"HeapReleasedBytes":2465792,"CGoAllocBytes":10114144,"CGoTotalBytes":19705856,"CGoCallRate":0.29998103,"CPUUserPercent":71.595474,"CPUSysPercent":6.2996016,"GCPausePercent":0.006505139,"GCRunCount":132,"NetHostRecvBytes":177080,"NetHostSendBytes":177080}
I211217 09:11:15.340847 18002 2@server/status/runtime_log.go:47  [n1] 1703  runtime stats: 1.2 GiB RSS, 486 goroutines (stacks: 5.2 MiB), 97 MiB/212 MiB Go alloc/total(stale) (heap fragmentation: 27 MiB, heap reserved: 64 MiB, heap released: 2.4 MiB), 9.6 MiB/19 MiB CGO alloc/total (0.3 CGO/sec), 71.6/6.3 %(u/s)time, 0.0 %gc (132x), 173 KiB/173 KiB (r/w)net
I211217 09:11:17.947707 23698 jobs/adopt.go:106  [nsql2] 1704  claimed 1 jobs
I211217 09:11:17.967348 25730 jobs/adopt.go:247  [nsql2] 1705  job 719787833294815233: resuming execution
I211217 09:11:18.028652 25643 jobs/registry.go:1179  [nsql2] 1706  MIGRATION job 719787833294815233: stepping through state running with error: <nil>
I211217 09:11:18.096396 25643 jobs/registry.go:1179  [nsql2] 1707  MIGRATION job 719787833294815233: stepping through state succeeded with error: <nil>
I211217 09:11:18.687822 23700 server/settingswatcher/settings_watcher.go:112  [nsql2,rangefeed=settings] 1708  set cluster version to: 21.2-34
I211217 09:11:18.687819 23977 server/settingswatcher/settings_watcher.go:112  [nsql3,rangefeed=settings] 1709  set cluster version to: 21.2-34
I211217 09:11:18.750238 23964 util/log/event_log.go:32  [nsql3,client=127.0.0.1:59434,hostssl,user=root] 1710 ={"Timestamp":1639732215323406553,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211217 09:11:18.856916 23744 2@rpc/context.go:743  [nsql3] 1711  closing [{n0: 127.0.0.1:33899 (default)}]
I211217 09:11:18.857543 23800 2@rpc/context.go:743  [nsql3] 1712  closing [{n0: 127.0.0.1:33899 (system)}]
I211217 09:11:18.858738 23744 2@rpc/context.go:743  [nsql3] 1713  closing [{n1: 127.0.0.1:33899 (default)}]
I211217 09:11:18.858806 23809 2@rpc/context.go:743  [nsql3] 1714  closing [{n1: 127.0.0.1:33899 (default)} {n0: 127.0.0.1:33899 (default)}]
I211217 09:11:18.859962 23536 2@rpc/context.go:743  [nsql2] 1715  closing [{n1: 127.0.0.1:33899 (default)} {n0: 127.0.0.1:33899 (default)}]
W211217 09:11:18.860674 23648 jobs/registry.go:851  [nsql2] 1716  canceling all adopted jobs due to stopper quiescing
I211217 09:11:18.860690 23578 2@rpc/context.go:743  [nsql2] 1717  closing [{n0: 127.0.0.1:33899 (system)}]
W211217 09:11:18.861391 23973 jobs/registry.go:851  [nsql3] 1718  canceling all adopted jobs due to stopper quiescing
W211217 09:11:18.861732 23618 ccl/kvccl/kvtenantccl/connector.go:182  [nsql2,tenant-connector] 1719  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211217 09:11:18.863050 23869 ccl/kvccl/kvtenantccl/connector.go:182  [nsql3,tenant-connector] 1720  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:60408->127.0.0.1:33899: use of closed network connection
W211217 09:11:18.864606 23885 sql/sqlliveness/slinstance/slinstance.go:222  [nsql3] 1721  exiting heartbeat loop
W211217 09:11:18.865041 23504 sql/sqlliveness/slinstance/slinstance.go:222  [nsql2] 1722  exiting heartbeat loop
    testing.go:1152: race detected during execution of test
I211217 09:11:18.872278 17713 testutils/testcluster/testcluster.go:106  [-] 1723  TestCluster quiescing nodes
I211217 09:11:18.874052 17878 kv/kvclient/kvcoord/transport_race.go:107  [-] 1724  transport race promotion: ran 233 iterations on up to 3075 requests
I211217 09:11:18.876428 17812 2@rpc/context.go:743  [n1] 1725  closing [{n1: 127.0.0.1:33899 (default)}]
W211217 09:11:18.876595 18021 sql/sqlliveness/slinstance/slinstance.go:222  [n1] 1726  exiting heartbeat loop
I211217 09:11:18.878501 18911 2@rpc/context.go:743  [n1] 1727  closing [{n1: 127.0.0.1:33899 (default)} {n0: 127.0.0.1:33899 (default)}]
W211217 09:11:18.879809 18044 jobs/registry.go:851  [n1] 1728  canceling all adopted jobs due to stopper quiescing
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (72.59s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3944536&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3944536&tab=artifacts#/) on master @ [3f95a4bd83cce2952a12497de82692866b4da659](https://github.com/cockroachdb/cockroach/commits/3f95a4bd83cce2952a12497de82692866b4da659):


```
I211218 10:01:02.444419 22910 2@util/log/event_log.go:32  [nsql3] 1682 ={"Timestamp":1639821662444401335,"EventType":"runtime_stats","MemRSSBytes":1331638272,"GoroutineCount":489,"MemStackSysBytes":6127616,"GoAllocBytes":126551232,"GoTotalBytes":214481960,"GoStatsStaleness":0.002561727,"HeapFragmentBytes":27491136,"HeapReservedBytes":37937152,"HeapReleasedBytes":3219456,"CGoAllocBytes":5938400,"CGoTotalBytes":15204352,"CGoCallRate":0.29998586,"CPUUserPercent":67.09683,"CPUSysPercent":5.3997455,"GCPausePercent":0.004385353,"GCRunCount":118,"NetHostRecvBytes":106014,"NetHostSendBytes":106014}
I211218 10:01:02.445920 22910 2@server/status/runtime_log.go:47  [nsql3] 1683  runtime stats: 1.2 GiB RSS, 489 goroutines (stacks: 5.8 MiB), 121 MiB/204 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 36 MiB, heap released: 3.1 MiB), 5.7 MiB/14 MiB CGO alloc/total (0.3 CGO/sec), 67.1/5.4 %(u/s)time, 0.0 %gc (118x), 104 KiB/104 KiB (r/w)net
I211218 10:01:03.644845 16941 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1684  [JOB 6] WAL created 000007
I211218 10:01:11.926419 22644 2@util/log/event_log.go:32  [nsql2] 1685 ={"Timestamp":1639821671926404798,"EventType":"runtime_stats","MemRSSBytes":1347473408,"GoroutineCount":489,"MemStackSysBytes":5767168,"GoAllocBytes":143762960,"GoTotalBytes":216153128,"GoStatsStaleness":0.001824273,"HeapFragmentBytes":27810288,"HeapReservedBytes":22437888,"HeapReleasedBytes":1548288,"CGoAllocBytes":10132832,"CGoTotalBytes":19402752,"CGoCallRate":0.3999632,"CPUUserPercent":63.59415,"CPUSysPercent":4.2996044,"GCPausePercent":0.0023326655,"GCRunCount":120,"NetHostRecvBytes":93774,"NetHostSendBytes":93774}
I211218 10:01:11.927534 22644 2@server/status/runtime_log.go:47  [nsql2] 1686  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.5 MiB), 137 MiB/206 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 21 MiB, heap released: 1.5 MiB), 9.7 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 63.6/4.3 %(u/s)time, 0.0 %gc (120x), 92 KiB/92 KiB (r/w)net
I211218 10:01:12.193567 16780 2@gossip/gossip.go:566  [n1] 1687  gossip status (ok, 1 node)
I211218 10:01:12.193567 16780 2@gossip/gossip.go:566  [n1] 1687 +gossip client (0/3 cur/max conns)
I211218 10:01:12.193567 16780 2@gossip/gossip.go:566  [n1] 1687 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211218 10:01:12.193567 16780 2@gossip/gossip.go:566  [n1] 1687 +gossip connectivity
I211218 10:01:12.193567 16780 2@gossip/gossip.go:566  [n1] 1687 +  n1 [sentinel];
I211218 10:01:12.199114 16785 2@util/log/event_log.go:32  [n1] 1688 ={"Timestamp":1639821672199103755,"EventType":"runtime_stats","MemRSSBytes":1345699840,"GoroutineCount":489,"MemStackSysBytes":6029312,"GoAllocBytes":148879240,"GoTotalBytes":216153128,"GoStatsStaleness":0.001911777,"HeapFragmentBytes":26937464,"HeapReservedBytes":17932288,"HeapReleasedBytes":1548288,"CGoAllocBytes":10132832,"CGoTotalBytes":19402752,"CGoCallRate":0.39998078,"CPUUserPercent":60.89707,"CPUSysPercent":4.099803,"GCPausePercent":0.002332768,"GCRunCount":120,"NetHostRecvBytes":90408,"NetHostSendBytes":90408}
I211218 10:01:12.200151 16785 2@server/status/runtime_log.go:47  [n1] 1689  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.8 MiB), 142 MiB/206 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 17 MiB, heap released: 1.5 MiB), 9.7 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 60.9/4.1 %(u/s)time, 0.0 %gc (120x), 88 KiB/88 KiB (r/w)net
I211218 10:01:12.444298 22910 2@util/log/event_log.go:32  [nsql3] 1690 ={"Timestamp":1639821672444277408,"EventType":"runtime_stats","MemRSSBytes":1353527296,"GoroutineCount":491,"MemStackSysBytes":6979584,"GoAllocBytes":158696808,"GoTotalBytes":216153128,"GoStatsStaleness":0.002058767,"HeapFragmentBytes":22854296,"HeapReservedBytes":11247616,"HeapReleasedBytes":1548288,"CGoAllocBytes":10132832,"CGoTotalBytes":19402752,"CGoCallRate":0.40000492,"CPUUserPercent":64.70079,"CPUSysPercent":4.5000553,"GCPausePercent":0.0023329086,"GCRunCount":120,"NetHostRecvBytes":91740,"NetHostSendBytes":91740}
I211218 10:01:12.445489 22910 2@server/status/runtime_log.go:47  [nsql3] 1691  runtime stats: 1.3 GiB RSS, 491 goroutines (stacks: 6.7 MiB), 151 MiB/206 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 11 MiB, heap released: 1.5 MiB), 9.7 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 64.7/4.5 %(u/s)time, 0.0 %gc (120x), 90 KiB/90 KiB (r/w)net
I211218 10:01:13.853282 17955 server/auto_upgrade.go:44  [n1] 1692  auto upgrade disabled by testing
I211218 10:01:21.927836 22644 2@util/log/event_log.go:32  [nsql2] 1693 ={"Timestamp":1639821681927820746,"EventType":"runtime_stats","MemRSSBytes":1363087360,"GoroutineCount":488,"MemStackSysBytes":5144576,"GoAllocBytes":119132984,"GoTotalBytes":217504808,"GoStatsStaleness":0.002090217,"HeapFragmentBytes":28634312,"HeapReservedBytes":48218112,"HeapReleasedBytes":196608,"CGoAllocBytes":10132832,"CGoTotalBytes":19402752,"CGoCallRate":0.29995763,"CPUUserPercent":72.189804,"CPUSysPercent":5.8991666,"GCPausePercent":0.0041251774,"GCRunCount":123,"NetHostRecvBytes":90361,"NetHostSendBytes":90361}
I211218 10:01:21.928996 22644 2@server/status/runtime_log.go:47  [nsql2] 1694  runtime stats: 1.3 GiB RSS, 488 goroutines (stacks: 4.9 MiB), 114 MiB/207 MiB Go alloc/total(stale) (heap fragmentation: 27 MiB, heap reserved: 46 MiB, heap released: 192 KiB), 9.7 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 72.2/5.9 %(u/s)time, 0.0 %gc (123x), 88 KiB/88 KiB (r/w)net
I211218 10:01:22.126522 24552 jobs/registry.go:1179  [nsql2] 1695  AUTO CREATE STATS job 720081000928903170: stepping through state running with error: <nil>
I211218 10:01:22.200477 16785 2@util/log/event_log.go:32  [n1] 1696 ={"Timestamp":1639821682200460050,"EventType":"runtime_stats","MemRSSBytes":1336573952,"GoroutineCount":494,"MemStackSysBytes":6324224,"GoAllocBytes":140785824,"GoTotalBytes":217504808,"GoStatsStaleness":0.002699235,"HeapFragmentBytes":24889184,"HeapReservedBytes":29130752,"HeapReleasedBytes":196608,"CGoAllocBytes":10132832,"CGoTotalBytes":19402752,"CGoCallRate":0.2999596,"CPUUserPercent":76.4897,"CPUSysPercent":6.599111,"GCPausePercent":0.0041252044,"GCRunCount":123,"NetHostRecvBytes":131132,"NetHostSendBytes":131132}
I211218 10:01:22.201869 16785 2@server/status/runtime_log.go:47  [n1] 1697  runtime stats: 1.2 GiB RSS, 494 goroutines (stacks: 6.0 MiB), 134 MiB/207 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 28 MiB, heap released: 192 KiB), 9.7 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 76.5/6.6 %(u/s)time, 0.0 %gc (123x), 128 KiB/128 KiB (r/w)net
I211218 10:01:22.288936 24552 jobs/registry.go:1179  [nsql2] 1698  AUTO CREATE STATS job 720081000928903170: stepping through state succeeded with error: <nil>
I211218 10:01:22.444855 22910 2@util/log/event_log.go:32  [nsql3] 1699 ={"Timestamp":1639821682444839597,"EventType":"runtime_stats","MemRSSBytes":1312337920,"GoroutineCount":488,"MemStackSysBytes":6193152,"GoAllocBytes":162919216,"GoTotalBytes":217504808,"GoStatsStaleness":0.002058587,"HeapFragmentBytes":22441168,"HeapReservedBytes":9576448,"HeapReleasedBytes":196608,"CGoAllocBytes":10132832,"CGoTotalBytes":19402752,"CGoCallRate":0.29998308,"CPUUserPercent":77.69562,"CPUSysPercent":6.599628,"GCPausePercent":0.004125527,"GCRunCount":123,"NetHostRecvBytes":161321,"NetHostSendBytes":161321}
I211218 10:01:22.446014 22910 2@server/status/runtime_log.go:47  [nsql3] 1700  runtime stats: 1.2 GiB RSS, 488 goroutines (stacks: 5.9 MiB), 155 MiB/207 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 9.1 MiB, heap released: 192 KiB), 9.7 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 77.7/6.6 %(u/s)time, 0.0 %gc (123x), 158 KiB/158 KiB (r/w)net
I211218 10:01:26.288779 22682 jobs/adopt.go:106  [nsql2] 1701  claimed 1 jobs
I211218 10:01:26.307687 24688 jobs/adopt.go:247  [nsql2] 1702  job 720080802779725825: resuming execution
I211218 10:01:26.356854 24774 jobs/registry.go:1179  [nsql2] 1703  MIGRATION job 720080802779725825: stepping through state running with error: <nil>
I211218 10:01:26.431057 24774 jobs/registry.go:1179  [nsql2] 1704  MIGRATION job 720080802779725825: stepping through state succeeded with error: <nil>
I211218 10:01:26.816343 22684 server/settingswatcher/settings_watcher.go:112  [nsql2,rangefeed=settings] 1705  set cluster version to: 21.2-34
I211218 10:01:26.816379 22999 server/settingswatcher/settings_watcher.go:112  [nsql3,rangefeed=settings] 1706  set cluster version to: 21.2-34
I211218 10:01:26.873336 23031 util/log/event_log.go:32  [nsql3,client=127.0.0.1:33226,hostssl,user=root] 1707 ={"Timestamp":1639821622574796729,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211218 10:01:26.974655 22851 2@rpc/context.go:743  [nsql3] 1708  closing [{n0: 127.0.0.1:43795 (system)}]
I211218 10:01:26.975289 22540 2@rpc/context.go:743  [nsql2] 1710  closing [{n0: 127.0.0.1:43795 (system)}]
W211218 10:01:26.976401 22995 jobs/registry.go:851  [nsql3] 1711  canceling all adopted jobs due to stopper quiescing
I211218 10:01:26.976676 22600 2@rpc/context.go:743  [nsql2] 1712  closing [{n1: 127.0.0.1:43795 (default)}]
I211218 10:01:26.974659 22964 2@rpc/context.go:743  [nsql3] 1713  closing [{n1: 127.0.0.1:43795 (default)} {n0: 127.0.0.1:43795 (default)}]
I211218 10:01:26.977030 22648 2@rpc/context.go:743  [nsql2] 1714  closing [{n1: 127.0.0.1:43795 (default)} {n0: 127.0.0.1:43795 (default)}]
I211218 10:01:26.974844 22796 2@rpc/context.go:743  [nsql3] 1709  closing [{n0: 127.0.0.1:43795 (system)}]
W211218 10:01:26.978125 22680 jobs/registry.go:851  [nsql2] 1715  canceling all adopted jobs due to stopper quiescing
W211218 10:01:26.978458 22558 sql/sqlliveness/slinstance/slinstance.go:222  [nsql2] 1716  exiting heartbeat loop
W211218 10:01:26.978976 22645 ccl/kvccl/kvtenantccl/connector.go:182  [nsql2,tenant-connector] 1717  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211218 10:01:26.979890 22833 sql/sqlliveness/slinstance/slinstance.go:222  [nsql3] 1718  exiting heartbeat loop
W211218 10:01:26.980932 22911 ccl/kvccl/kvtenantccl/connector.go:182  [nsql3,tenant-connector] 1719  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:37114->127.0.0.1:43795: use of closed network connection
    testing.go:1152: race detected during execution of test
I211218 10:01:26.988272 16736 testutils/testcluster/testcluster.go:106  [-] 1720  TestCluster quiescing nodes
I211218 10:01:26.989444 16746 kv/kvclient/kvcoord/transport_race.go:107  [-] 1721  transport race promotion: ran 211 iterations on up to 3072 requests
W211218 10:01:26.994892 17060 sql/sqlliveness/slinstance/slinstance.go:222  [n1] 1722  exiting heartbeat loop
I211218 10:01:26.995048 16817 2@rpc/context.go:743  [n1] 1723  closing [{n1: 127.0.0.1:43795 (default)}]
I211218 10:01:26.995341 17930 2@rpc/context.go:743  [n1] 1724  closing [{n1: 127.0.0.1:43795 (default)} {n0: 127.0.0.1:43795 (default)}]
W211218 10:01:26.995040 17082 jobs/registry.go:851  [n1] 1725  canceling all adopted jobs due to stopper quiescing
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (73.72s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3949850&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3949850&tab=artifacts#/) on master @ [4cbe795c7f8c36c007c3ca1250367cb181f5ca0c](https://github.com/cockroachdb/cockroach/commits/4cbe795c7f8c36c007c3ca1250367cb181f5ca0c):


```
I211219 09:00:55.120039 23720 2@util/log/event_log.go:32  [nsql3] 1962 ={"Timestamp":1639904455120023628,"EventType":"runtime_stats","MemRSSBytes":956891136,"GoroutineCount":489,"MemStackSysBytes":5537792,"GoAllocBytes":77256720,"GoTotalBytes":139546648,"GoStatsStaleness":0.001806113,"HeapFragmentBytes":18892784,"HeapReservedBytes":24666112,"HeapReleasedBytes":3670016,"CGoAllocBytes":6081760,"CGoTotalBytes":14618624,"CGoCallRate":0.2999942,"CPUUserPercent":64.99874,"CPUSysPercent":3.8999245,"GCPausePercent":0.005114571,"GCRunCount":147,"NetHostRecvBytes":109989,"NetHostSendBytes":109989}
I211219 09:00:55.121041 23720 2@server/status/runtime_log.go:47  [nsql3] 1963  runtime stats: 913 MiB RSS, 489 goroutines (stacks: 5.3 MiB), 74 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 24 MiB, heap released: 3.5 MiB), 5.8 MiB/14 MiB CGO alloc/total (0.3 CGO/sec), 65.0/3.9 %(u/s)time, 0.0 %gc (147x), 107 KiB/107 KiB (r/w)net
I211219 09:00:55.830831 17796 2@util/log/event_log.go:32  [n1] 1964 ={"Timestamp":1639904455830818170,"EventType":"runtime_stats","MemRSSBytes":954429440,"GoroutineCount":489,"MemStackSysBytes":5898240,"GoAllocBytes":92064592,"GoTotalBytes":139546648,"GoStatsStaleness":0.001970831,"HeapFragmentBytes":17601712,"HeapReservedBytes":10788864,"HeapReleasedBytes":3670016,"CGoAllocBytes":6081760,"CGoTotalBytes":14618624,"CGoCallRate":0.29996657,"CPUUserPercent":65.892654,"CPUSysPercent":3.7995768,"GCPausePercent":0.0051141,"GCRunCount":147,"NetHostRecvBytes":107624,"NetHostSendBytes":107624}
I211219 09:00:55.831891 17796 2@server/status/runtime_log.go:47  [n1] 1965  runtime stats: 910 MiB RSS, 489 goroutines (stacks: 5.6 MiB), 88 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 10 MiB, heap released: 3.5 MiB), 5.8 MiB/14 MiB CGO alloc/total (0.3 CGO/sec), 65.9/3.8 %(u/s)time, 0.0 %gc (147x), 105 KiB/105 KiB (r/w)net
I211219 09:00:57.108781 17746 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1966  [JOB 6] WAL created 000007
I211219 09:01:04.630588 23408 2@util/log/event_log.go:32  [nsql2] 1967 ={"Timestamp":1639904464630572173,"EventType":"runtime_stats","MemRSSBytes":973930496,"GoroutineCount":489,"MemStackSysBytes":5308416,"GoAllocBytes":69106000,"GoTotalBytes":141053976,"GoStatsStaleness":0.001811817,"HeapFragmentBytes":19515056,"HeapReservedBytes":33931264,"HeapReleasedBytes":2162688,"CGoAllocBytes":10276192,"CGoTotalBytes":18817024,"CGoCallRate":0.40000308,"CPUUserPercent":62.000477,"CPUSysPercent":4.300033,"GCPausePercent":0.0057736947,"GCRunCount":151,"NetHostRecvBytes":92311,"NetHostSendBytes":92311}
I211219 09:01:04.631664 23408 2@server/status/runtime_log.go:47  [nsql2] 1968  runtime stats: 929 MiB RSS, 489 goroutines (stacks: 5.1 MiB), 66 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 19 MiB, heap reserved: 32 MiB, heap released: 2.1 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 62.0/4.3 %(u/s)time, 0.0 %gc (151x), 90 KiB/90 KiB (r/w)net
I211219 09:01:05.121076 23720 2@util/log/event_log.go:32  [nsql3] 1969 ={"Timestamp":1639904465121062268,"EventType":"runtime_stats","MemRSSBytes":971542528,"GoroutineCount":489,"MemStackSysBytes":5472256,"GoAllocBytes":81330592,"GoTotalBytes":141053976,"GoStatsStaleness":0.001835864,"HeapFragmentBytes":18497120,"HeapReservedBytes":22560768,"HeapReleasedBytes":2162688,"CGoAllocBytes":10276192,"CGoTotalBytes":18817024,"CGoCallRate":0.39995846,"CPUUserPercent":61.793583,"CPUSysPercent":4.3995433,"GCPausePercent":0.0057730507,"GCRunCount":151,"NetHostRecvBytes":90596,"NetHostSendBytes":90596}
I211219 09:01:05.122060 23720 2@server/status/runtime_log.go:47  [nsql3] 1970  runtime stats: 926 MiB RSS, 489 goroutines (stacks: 5.2 MiB), 78 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 22 MiB, heap released: 2.1 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 61.8/4.4 %(u/s)time, 0.0 %gc (151x), 88 KiB/88 KiB (r/w)net
I211219 09:01:05.823600 17537 2@gossip/gossip.go:566  [n1] 1971  gossip status (ok, 1 node)
I211219 09:01:05.823600 17537 2@gossip/gossip.go:566  [n1] 1971 +gossip client (0/3 cur/max conns)
I211219 09:01:05.823600 17537 2@gossip/gossip.go:566  [n1] 1971 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211219 09:01:05.823600 17537 2@gossip/gossip.go:566  [n1] 1971 +gossip connectivity
I211219 09:01:05.823600 17537 2@gossip/gossip.go:566  [n1] 1971 +  n1 [sentinel];
I211219 09:01:05.832122 17796 2@util/log/event_log.go:32  [n1] 1972 ={"Timestamp":1639904465832107785,"EventType":"runtime_stats","MemRSSBytes":970235904,"GoroutineCount":489,"MemStackSysBytes":4751360,"GoAllocBytes":53826744,"GoTotalBytes":141176856,"GoStatsStaleness":0.002900074,"HeapFragmentBytes":21621576,"HeapReservedBytes":47783936,"HeapReleasedBytes":2039808,"CGoAllocBytes":10276192,"CGoTotalBytes":18817024,"CGoCallRate":0.39994854,"CPUUserPercent":63.091885,"CPUSysPercent":4.499421,"GCPausePercent":0.0080014,"GCRunCount":152,"NetHostRecvBytes":92153,"NetHostSendBytes":92153}
I211219 09:01:05.833873 17796 2@server/status/runtime_log.go:47  [n1] 1973  runtime stats: 925 MiB RSS, 489 goroutines (stacks: 4.5 MiB), 51 MiB/135 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 46 MiB, heap released: 1.9 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 63.1/4.5 %(u/s)time, 0.0 %gc (152x), 90 KiB/90 KiB (r/w)net
I211219 09:01:06.316792 18744 server/auto_upgrade.go:44  [n1] 1974  auto upgrade disabled by testing
I211219 09:01:14.631128 23408 2@util/log/event_log.go:32  [nsql2] 1975 ={"Timestamp":1639904474631110869,"EventType":"runtime_stats","MemRSSBytes":978112512,"GoroutineCount":488,"MemStackSysBytes":5505024,"GoAllocBytes":84797600,"GoTotalBytes":144674840,"GoStatsStaleness":0.001931551,"HeapFragmentBytes":19183456,"HeapReservedBytes":21995520,"HeapReleasedBytes":2736128,"CGoAllocBytes":10276192,"CGoTotalBytes":18817024,"CGoCallRate":0.2999838,"CPUUserPercent":66.796394,"CPUSysPercent":5.3997083,"GCPausePercent":0.010980007,"GCRunCount":155,"NetHostRecvBytes":91993,"NetHostSendBytes":91993}
I211219 09:01:14.632253 23408 2@server/status/runtime_log.go:47  [nsql2] 1976  runtime stats: 933 MiB RSS, 488 goroutines (stacks: 5.3 MiB), 81 MiB/138 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 21 MiB, heap released: 2.6 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 66.8/5.4 %(u/s)time, 0.0 %gc (155x), 90 KiB/90 KiB (r/w)net
I211219 09:01:14.816155 25426 jobs/registry.go:1179  [nsql2] 1977  AUTO CREATE STATS job 720352296066678786: stepping through state running with error: <nil>
I211219 09:01:14.864376 23491 jobs/adopt.go:106  [nsql2] 1978  claimed 1 jobs
I211219 09:01:14.887496 25376 jobs/adopt.go:247  [nsql2] 1979  job 720352097992835073: resuming execution
I211219 09:01:14.953099 25359 jobs/registry.go:1179  [nsql2] 1980  MIGRATION job 720352097992835073: stepping through state running with error: <nil>
I211219 09:01:14.972489 25426 jobs/registry.go:1179  [nsql2] 1981  AUTO CREATE STATS job 720352296066678786: stepping through state succeeded with error: <nil>
I211219 09:01:15.032335 25359 jobs/registry.go:1179  [nsql2] 1982  MIGRATION job 720352097992835073: stepping through state succeeded with error: <nil>
I211219 09:01:15.122222 23720 2@util/log/event_log.go:32  [nsql3] 1983 ={"Timestamp":1639904475122208505,"EventType":"runtime_stats","MemRSSBytes":909381632,"GoroutineCount":487,"MemStackSysBytes":5537792,"GoAllocBytes":89835512,"GoTotalBytes":144936984,"GoStatsStaleness":0.002222551,"HeapFragmentBytes":16070664,"HeapReservedBytes":20037632,"HeapReleasedBytes":2736128,"CGoAllocBytes":10276192,"CGoTotalBytes":18817024,"CGoCallRate":0.29996562,"CPUUserPercent":73.89153,"CPUSysPercent":6.3992662,"GCPausePercent":0.012605275,"GCRunCount":156,"NetHostRecvBytes":202680,"NetHostSendBytes":202680}
I211219 09:01:15.123286 23720 2@server/status/runtime_log.go:47  [nsql3] 1984  runtime stats: 867 MiB RSS, 487 goroutines (stacks: 5.3 MiB), 86 MiB/138 MiB Go alloc/total(stale) (heap fragmentation: 15 MiB, heap reserved: 19 MiB, heap released: 2.6 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 73.9/6.4 %(u/s)time, 0.0 %gc (156x), 198 KiB/198 KiB (r/w)net
I211219 09:01:15.672336 23772 server/settingswatcher/settings_watcher.go:112  [nsql3,rangefeed=settings] 1985  set cluster version to: 21.2-34
I211219 09:01:15.672733 23493 server/settingswatcher/settings_watcher.go:112  [nsql2,rangefeed=settings] 1986  set cluster version to: 21.2-34
I211219 09:01:15.722170 23755 util/log/event_log.go:32  [nsql3,client=127.0.0.1:45298,hostssl,user=root] 1987 ={"Timestamp":1639904415249437952,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211219 09:01:15.812156 23548 2@rpc/context.go:743  [nsql3] 1988  closing [{n0: 127.0.0.1:46627 (default)}]
W211219 09:01:15.812632 23768 jobs/registry.go:851  [nsql3] 1989  canceling all adopted jobs due to stopper quiescing
I211219 09:01:15.813453 23548 2@rpc/context.go:743  [nsql3] 1990  closing [{n0: 127.0.0.1:46627 (system)}]
I211219 09:01:15.813716 23698 2@rpc/context.go:743  [nsql3] 1991  closing [{n0: 127.0.0.1:46627 (system)}]
W211219 09:01:15.813906 23433 sql/sqlliveness/slinstance/slinstance.go:222  [nsql2] 1992  exiting heartbeat loop
I211219 09:01:15.814772 23746 2@rpc/context.go:743  [nsql3] 1993  closing [{n1: 127.0.0.1:46627 (default)} {n0: 127.0.0.1:46627 (default)}]
I211219 09:01:15.815139 23334 2@rpc/context.go:743  [nsql2] 1994  closing [{n0: 127.0.0.1:46627 (default)}]
W211219 09:01:15.815395 23645 sql/sqlliveness/slinstance/slinstance.go:222  [nsql3] 1995  exiting heartbeat loop
I211219 09:01:15.815430 23412 2@rpc/context.go:743  [nsql2] 1996  closing [{n0: 127.0.0.1:46627 (system)}]
W211219 09:01:15.816099 23473 jobs/registry.go:851  [nsql2] 1997  canceling all adopted jobs due to stopper quiescing
W211219 09:01:15.817265 23721 ccl/kvccl/kvtenantccl/connector.go:182  [nsql3,tenant-connector] 1998  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:53578->127.0.0.1:46627: use of closed network connection
W211219 09:01:15.817589 23409 ccl/kvccl/kvtenantccl/connector.go:182  [nsql2,tenant-connector] 1999  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:53572->127.0.0.1:46627: use of closed network connection
I211219 09:01:15.815139 23419 2@rpc/context.go:743  [nsql2] 2000  closing [{n1: 127.0.0.1:46627 (default)} {n0: 127.0.0.1:46627 (default)}]
    testing.go:1152: race detected during execution of test
I211219 09:01:15.826721 16479 testutils/testcluster/testcluster.go:106  [-] 2001  TestCluster quiescing nodes
W211219 09:01:15.827475 17922 jobs/registry.go:851  [n1] 2002  canceling all adopted jobs due to stopper quiescing
I211219 09:01:15.827484 18760 2@rpc/context.go:743  [n1] 2003  closing [{n1: 127.0.0.1:46627 (default)} {n0: 127.0.0.1:46627 (default)}]
W211219 09:01:15.830801 17838 sql/sqlliveness/slinstance/slinstance.go:222  [n1] 2004  exiting heartbeat loop
I211219 09:01:16.121411 17831 kv/kvclient/kvcoord/transport_race.go:107  [-] 2005  transport race promotion: ran 221 iterations on up to 3060 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (69.05s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3954767&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3954767&tab=artifacts#/) on master @ [93c9a3bc0e234c23df45aee62f6f76eb3b5ab723](https://github.com/cockroachdb/cockroach/commits/93c9a3bc0e234c23df45aee62f6f76eb3b5ab723):


```
I211220 09:24:59.752443 17558 2@util/log/event_log.go:32  [n1] 1685 ={"Timestamp":1639992299752420476,"EventType":"runtime_stats","MemRSSBytes":1330245632,"GoroutineCount":489,"MemStackSysBytes":6160384,"GoAllocBytes":141929640,"GoTotalBytes":212605992,"GoStatsStaleness":0.002568239,"HeapFragmentBytes":24474456,"HeapReservedBytes":23404544,"HeapReleasedBytes":1163264,"CGoAllocBytes":5881056,"CGoTotalBytes":12308480,"CGoCallRate":0.2999431,"CPUUserPercent":65.78752,"CPUSysPercent":5.2989945,"GCPausePercent":0.0018323225,"GCRunCount":135,"NetHostRecvBytes":106386,"NetHostSendBytes":106386}
I211220 09:24:59.754068 17558 2@server/status/runtime_log.go:47  [n1] 1686  runtime stats: 1.2 GiB RSS, 489 goroutines (stacks: 5.9 MiB), 135 MiB/203 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 22 MiB, heap released: 1.1 MiB), 5.6 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 65.8/5.3 %(u/s)time, 0.0 %gc (135x), 104 KiB/104 KiB (r/w)net
I211220 09:25:01.023140 17477 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1687  [JOB 6] WAL created 000007
I211220 09:25:08.296827 18462 server/auto_upgrade.go:44  [n1] 1688  auto upgrade disabled by testing
I211220 09:25:08.543967 23042 2@util/log/event_log.go:32  [nsql2] 1689 ={"Timestamp":1639992308543952881,"EventType":"runtime_stats","MemRSSBytes":1345941504,"GoroutineCount":489,"MemStackSysBytes":5832704,"GoAllocBytes":147907504,"GoTotalBytes":215710760,"GoStatsStaleness":0.001738139,"HeapFragmentBytes":25427024,"HeapReservedBytes":19906560,"HeapReleasedBytes":2252800,"CGoAllocBytes":10075488,"CGoTotalBytes":16502784,"CGoCallRate":0.3999952,"CPUUserPercent":63.89923,"CPUSysPercent":4.399947,"GCPausePercent":0.0029452848,"GCRunCount":137,"NetHostRecvBytes":102328,"NetHostSendBytes":102328}
I211220 09:25:08.544944 23042 2@server/status/runtime_log.go:47  [nsql2] 1690  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.6 MiB), 141 MiB/206 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 19 MiB, heap released: 2.1 MiB), 9.6 MiB/16 MiB CGO alloc/total (0.4 CGO/sec), 63.9/4.4 %(u/s)time, 0.0 %gc (137x), 100 KiB/100 KiB (r/w)net
I211220 09:25:09.021610 23324 2@util/log/event_log.go:32  [nsql3] 1691 ={"Timestamp":1639992309021594835,"EventType":"runtime_stats","MemRSSBytes":1350336512,"GoroutineCount":489,"MemStackSysBytes":5931008,"GoAllocBytes":160051864,"GoTotalBytes":215710760,"GoStatsStaleness":0.001752769,"HeapFragmentBytes":24350056,"HeapReservedBytes":8740864,"HeapReleasedBytes":2252800,"CGoAllocBytes":10075488,"CGoTotalBytes":16502784,"CGoCallRate":0.39998353,"CPUUserPercent":63.497383,"CPUSysPercent":4.6998067,"GCPausePercent":0.0029451987,"GCRunCount":137,"NetHostRecvBytes":94024,"NetHostSendBytes":94024}
I211220 09:25:09.022642 23324 2@server/status/runtime_log.go:47  [nsql3] 1692  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.7 MiB), 153 MiB/206 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 8.3 MiB, heap released: 2.1 MiB), 9.6 MiB/16 MiB CGO alloc/total (0.4 CGO/sec), 63.5/4.7 %(u/s)time, 0.0 %gc (137x), 92 KiB/92 KiB (r/w)net
I211220 09:25:09.745105 17249 2@gossip/gossip.go:566  [n1] 1693  gossip status (ok, 1 node)
I211220 09:25:09.745105 17249 2@gossip/gossip.go:566  [n1] 1693 +gossip client (0/3 cur/max conns)
I211220 09:25:09.745105 17249 2@gossip/gossip.go:566  [n1] 1693 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211220 09:25:09.745105 17249 2@gossip/gossip.go:566  [n1] 1693 +gossip connectivity
I211220 09:25:09.745105 17249 2@gossip/gossip.go:566  [n1] 1693 +  n1 [sentinel];
I211220 09:25:09.752283 17558 2@util/log/event_log.go:32  [n1] 1694 ={"Timestamp":1639992309752273305,"EventType":"runtime_stats","MemRSSBytes":1343713280,"GoroutineCount":489,"MemStackSysBytes":5406720,"GoAllocBytes":94320128,"GoTotalBytes":215710760,"GoStatsStaleness":0.001699518,"HeapFragmentBytes":29272576,"HeapReservedBytes":70074368,"HeapReleasedBytes":2252800,"CGoAllocBytes":10075488,"CGoTotalBytes":16502784,"CGoCallRate":0.40000546,"CPUUserPercent":62.80086,"CPUSysPercent":4.5000615,"GCPausePercent":0.00439668,"GCRunCount":138,"NetHostRecvBytes":91643,"NetHostSendBytes":91643}
I211220 09:25:09.753241 17558 2@server/status/runtime_log.go:47  [n1] 1695  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.2 MiB), 90 MiB/206 MiB Go alloc/total(stale) (heap fragmentation: 28 MiB, heap reserved: 67 MiB, heap released: 2.1 MiB), 9.6 MiB/16 MiB CGO alloc/total (0.4 CGO/sec), 62.8/4.5 %(u/s)time, 0.0 %gc (138x), 90 KiB/90 KiB (r/w)net
I211220 09:25:18.544218 23042 2@util/log/event_log.go:32  [nsql2] 1696 ={"Timestamp":1639992318544205247,"EventType":"runtime_stats","MemRSSBytes":1373884416,"GoroutineCount":489,"MemStackSysBytes":5373952,"GoAllocBytes":109868112,"GoTotalBytes":221276216,"GoStatsStaleness":0.001774993,"HeapFragmentBytes":27372464,"HeapReservedBytes":59858944,"HeapReleasedBytes":3047424,"CGoAllocBytes":10075488,"CGoTotalBytes":16502784,"CGoCallRate":0.2999925,"CPUUserPercent":67.498314,"CPUSysPercent":5.19987,"GCPausePercent":0.003317527,"GCRunCount":140,"NetHostRecvBytes":90577,"NetHostSendBytes":90577}
I211220 09:25:18.545316 23042 2@server/status/runtime_log.go:47  [nsql2] 1697  runtime stats: 1.3 GiB RSS, 489 goroutines (stacks: 5.1 MiB), 105 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 57 MiB, heap released: 2.9 MiB), 9.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 67.5/5.2 %(u/s)time, 0.0 %gc (140x), 88 KiB/88 KiB (r/w)net
I211220 09:25:18.732165 24985 jobs/registry.go:1179  [nsql2] 1698  AUTO CREATE STATS job 720640142979923970: stepping through state running with error: <nil>
I211220 09:25:18.872928 24985 jobs/registry.go:1179  [nsql2] 1699  AUTO CREATE STATS job 720640142979923970: stepping through state succeeded with error: <nil>
I211220 09:25:19.022159 23324 2@util/log/event_log.go:32  [nsql3] 1700 ={"Timestamp":1639992319022144814,"EventType":"runtime_stats","MemRSSBytes":1328291840,"GoroutineCount":487,"MemStackSysBytes":5865472,"GoAllocBytes":151708248,"GoTotalBytes":221276216,"GoStatsStaleness":0.001755328,"HeapFragmentBytes":21822888,"HeapReservedBytes":23076864,"HeapReleasedBytes":3047424,"CGoAllocBytes":10075488,"CGoTotalBytes":16502784,"CGoCallRate":0.2999835,"CPUUserPercent":74.595894,"CPUSysPercent":5.399703,"GCPausePercent":0.0033174274,"GCRunCount":140,"NetHostRecvBytes":164197,"NetHostSendBytes":164197}
I211220 09:25:19.023148 23324 2@server/status/runtime_log.go:47  [nsql3] 1701  runtime stats: 1.2 GiB RSS, 487 goroutines (stacks: 5.6 MiB), 145 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 22 MiB, heap released: 2.9 MiB), 9.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 74.6/5.4 %(u/s)time, 0.0 %gc (140x), 160 KiB/160 KiB (r/w)net
I211220 09:25:19.754009 17558 2@util/log/event_log.go:32  [n1] 1702 ={"Timestamp":1639992319753995295,"EventType":"runtime_stats","MemRSSBytes":1314521088,"GoroutineCount":486,"MemStackSysBytes":5275648,"GoAllocBytes":96213240,"GoTotalBytes":221276216,"GoStatsStaleness":0.002517877,"HeapFragmentBytes":28329736,"HeapReservedBytes":72654848,"HeapReleasedBytes":3047424,"CGoAllocBytes":10075488,"CGoTotalBytes":16502784,"CGoCallRate":0.29994848,"CPUUserPercent":76.2869,"CPUSysPercent":5.898987,"GCPausePercent":0.0032212068,"GCRunCount":141,"NetHostRecvBytes":178243,"NetHostSendBytes":178243}
I211220 09:25:19.755613 17558 2@server/status/runtime_log.go:47  [n1] 1703  runtime stats: 1.2 GiB RSS, 486 goroutines (stacks: 5.0 MiB), 92 MiB/211 MiB Go alloc/total(stale) (heap fragmentation: 27 MiB, heap reserved: 69 MiB, heap released: 2.9 MiB), 9.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 76.3/5.9 %(u/s)time, 0.0 %gc (141x), 174 KiB/174 KiB (r/w)net
I211220 09:25:20.897969 23428 jobs/adopt.go:106  [nsql3] 1704  claimed 1 jobs
I211220 09:25:20.919024 25177 jobs/adopt.go:247  [nsql3] 1705  job 720639944903688193: resuming execution
I211220 09:25:20.977027 25180 jobs/registry.go:1179  [nsql3] 1706  MIGRATION job 720639944903688193: stepping through state running with error: <nil>
I211220 09:25:21.011515 25180 jobs/registry.go:1179  [nsql3] 1707  MIGRATION job 720639944903688193: stepping through state succeeded with error: <nil>
I211220 09:25:21.314256 23112 server/settingswatcher/settings_watcher.go:112  [nsql2,rangefeed=settings] 1708  set cluster version to: 21.2-34
I211220 09:25:21.314271 23430 server/settingswatcher/settings_watcher.go:112  [nsql3,rangefeed=settings] 1709  set cluster version to: 21.2-34
I211220 09:25:21.364156 23479 util/log/event_log.go:32  [nsql3,client=127.0.0.1:41228,hostssl,user=root] 1710 ={"Timestamp":1639992259146289403,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211220 09:25:21.449089 23146 2@rpc/context.go:743  [nsql3] 1711  closing [{n0: 127.0.0.1:32915 (system)}]
W211220 09:25:21.449903 23108 jobs/registry.go:851  [nsql2] 1712  canceling all adopted jobs due to stopper quiescing
I211220 09:25:21.450080 23146 2@rpc/context.go:743  [nsql3] 1713  closing [{n1: 127.0.0.1:32915 (default)}]
W211220 09:25:21.450175 23426 jobs/registry.go:851  [nsql3] 1714  canceling all adopted jobs due to stopper quiescing
I211220 09:25:21.450480 23064 2@rpc/context.go:743  [nsql2] 1715  closing [{n1: 127.0.0.1:32915 (default)} {n0: 127.0.0.1:32915 (default)}]
I211220 09:25:21.450744 23262 2@rpc/context.go:743  [nsql3] 1716  closing [{n1: 127.0.0.1:32915 (default)} {n0: 127.0.0.1:32915 (default)}]
I211220 09:25:21.451395 23033 2@rpc/context.go:743  [nsql2] 1717  closing [{n0: 127.0.0.1:32915 (system)}]
I211220 09:25:21.451722 23248 2@rpc/context.go:743  [nsql3] 1718  closing [{n0: 127.0.0.1:32915 (system)}]
W211220 09:25:21.453311 22971 sql/sqlliveness/slinstance/slinstance.go:222  [nsql2] 1719  exiting heartbeat loop
W211220 09:25:21.453481 23325 ccl/kvccl/kvtenantccl/connector.go:182  [nsql3,tenant-connector] 1720  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:39446->127.0.0.1:32915: use of closed network connection
W211220 09:25:21.454487 23329 sql/sqlliveness/slinstance/slinstance.go:222  [nsql3] 1722  exiting heartbeat loop
W211220 09:25:21.454328 23043 ccl/kvccl/kvtenantccl/connector.go:182  [nsql2,tenant-connector] 1721  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
    testing.go:1152: race detected during execution of test
I211220 09:25:21.463722 17274 testutils/testcluster/testcluster.go:106  [-] 1723  TestCluster quiescing nodes
I211220 09:25:21.467716 17341 kv/kvclient/kvcoord/transport_race.go:107  [-] 1724  transport race promotion: ran 224 iterations on up to 3046 requests
I211220 09:25:21.469231 17366 2@rpc/context.go:743  [n1] 1725  closing [{n1: 127.0.0.1:32915 (default)}]
W211220 09:25:21.470223 17611 jobs/registry.go:851  [n1] 1726  canceling all adopted jobs due to stopper quiescing
I211220 09:25:21.470467 18472 2@rpc/context.go:743  [n1] 1727  closing [{n1: 127.0.0.1:32915 (default)} {n0: 127.0.0.1:32915 (default)}]
W211220 09:25:21.471083 17567 sql/sqlliveness/slinstance/slinstance.go:222  [n1] 1728  exiting heartbeat loop
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (70.78s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3964818&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3964818&tab=artifacts#/) on master @ [2bf850847554519dc139289615e1a061b6a251de](https://github.com/cockroachdb/cockroach/commits/2bf850847554519dc139289615e1a061b6a251de):


```
I211221 10:27:43.466999 21131 2@server/status/runtime_log.go:47  [nsql3] 1687  runtime stats: 980 MiB RSS, 489 goroutines (stacks: 4.8 MiB), 52 MiB/140 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 45 MiB, heap released: 19 MiB), 5.8 MiB/14 MiB CGO alloc/total (0.3 CGO/sec), 74.8/5.0 %(u/s)time, 0.0 %gc (117x), 101 KiB/101 KiB (r/w)net
I211221 10:27:44.253597 15178 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1688  [JOB 6] WAL created 000007
I211221 10:27:52.889136 15211 2@gossip/gossip.go:566  [n1] 1689  gossip status (ok, 1 node)
I211221 10:27:52.889136 15211 2@gossip/gossip.go:566  [n1] 1689 +gossip client (0/3 cur/max conns)
I211221 10:27:52.889136 15211 2@gossip/gossip.go:566  [n1] 1689 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211221 10:27:52.889136 15211 2@gossip/gossip.go:566  [n1] 1689 +gossip connectivity
I211221 10:27:52.889136 15211 2@gossip/gossip.go:566  [n1] 1689 +  n1 [sentinel];
I211221 10:27:52.896021 15216 2@util/log/event_log.go:32  [n1] 1690 ={"Timestamp":1640082472896007359,"EventType":"runtime_stats","MemRSSBytes":1034170368,"GoroutineCount":489,"MemStackSysBytes":5668864,"GoAllocBytes":86751616,"GoTotalBytes":149453864,"GoStatsStaleness":0.002329339,"HeapFragmentBytes":21694080,"HeapReservedBytes":19619840,"HeapReleasedBytes":17260544,"CGoAllocBytes":10247520,"CGoTotalBytes":18874368,"CGoCallRate":0.39996704,"CPUUserPercent":67.59444,"CPUSysPercent":4.899596,"GCPausePercent":0.008327744,"GCRunCount":120,"NetHostRecvBytes":91576,"NetHostSendBytes":91576}
I211221 10:27:52.897481 15216 2@server/status/runtime_log.go:47  [n1] 1691  runtime stats: 986 MiB RSS, 489 goroutines (stacks: 5.4 MiB), 83 MiB/142 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 19 MiB, heap released: 16 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 67.6/4.9 %(u/s)time, 0.0 %gc (120x), 89 KiB/89 KiB (r/w)net
I211221 10:27:52.905230 20821 2@util/log/event_log.go:32  [nsql2] 1692 ={"Timestamp":1640082472905218725,"EventType":"runtime_stats","MemRSSBytes":1034362880,"GoroutineCount":491,"MemStackSysBytes":5799936,"GoAllocBytes":87418176,"GoTotalBytes":149453864,"GoStatsStaleness":0.001783038,"HeapFragmentBytes":21109440,"HeapReservedBytes":19406848,"HeapReleasedBytes":17260544,"CGoAllocBytes":10247520,"CGoTotalBytes":18874368,"CGoCallRate":0.39996403,"CPUUserPercent":67.59392,"CPUSysPercent":4.8995595,"GCPausePercent":0.008327682,"GCRunCount":120,"NetHostRecvBytes":91576,"NetHostSendBytes":91576}
I211221 10:27:52.906250 20821 2@server/status/runtime_log.go:47  [nsql2] 1693  runtime stats: 986 MiB RSS, 491 goroutines (stacks: 5.5 MiB), 83 MiB/142 MiB Go alloc/total(stale) (heap fragmentation: 20 MiB, heap reserved: 18 MiB, heap released: 16 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 67.6/4.9 %(u/s)time, 0.0 %gc (120x), 89 KiB/89 KiB (r/w)net
I211221 10:27:53.467127 21131 2@util/log/event_log.go:32  [nsql3] 1694 ={"Timestamp":1640082473467111920,"EventType":"runtime_stats","MemRSSBytes":1032523776,"GoroutineCount":490,"MemStackSysBytes":5537792,"GoAllocBytes":60886328,"GoTotalBytes":148552744,"GoStatsStaleness":0.001853592,"HeapFragmentBytes":22737608,"HeapReservedBytes":43671552,"HeapReleasedBytes":18161664,"CGoAllocBytes":10247520,"CGoTotalBytes":18874368,"CGoCallRate":0.39995265,"CPUUserPercent":67.691986,"CPUSysPercent":4.799432,"GCPausePercent":0.007943059,"GCRunCount":121,"NetHostRecvBytes":89893,"NetHostSendBytes":89893}
I211221 10:27:53.468144 21131 2@server/status/runtime_log.go:47  [nsql3] 1695  runtime stats: 985 MiB RSS, 490 goroutines (stacks: 5.3 MiB), 58 MiB/142 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 42 MiB, heap released: 17 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 67.7/4.8 %(u/s)time, 0.0 %gc (121x), 88 KiB/88 KiB (r/w)net
I211221 10:27:54.148941 16080 server/auto_upgrade.go:44  [n1] 1696  auto upgrade disabled by testing
I211221 10:28:02.896234 15216 2@util/log/event_log.go:32  [n1] 1697 ={"Timestamp":1640082482896223174,"EventType":"runtime_stats","MemRSSBytes":1038807040,"GoroutineCount":488,"MemStackSysBytes":4882432,"GoAllocBytes":67810392,"GoTotalBytes":151862312,"GoStatsStaleness":0.001827579,"HeapFragmentBytes":22563752,"HeapReservedBytes":40886272,"HeapReleasedBytes":14852096,"CGoAllocBytes":10247520,"CGoTotalBytes":18874368,"CGoCallRate":0.2999935,"CPUUserPercent":74.0984,"CPUSysPercent":5.2998853,"GCPausePercent":0.0057926145,"GCRunCount":125,"NetHostRecvBytes":93855,"NetHostSendBytes":93855}
I211221 10:28:02.897238 15216 2@server/status/runtime_log.go:47  [n1] 1698  runtime stats: 991 MiB RSS, 488 goroutines (stacks: 4.7 MiB), 65 MiB/145 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 39 MiB, heap released: 14 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 74.1/5.3 %(u/s)time, 0.0 %gc (125x), 92 KiB/92 KiB (r/w)net
I211221 10:28:02.905318 20821 2@util/log/event_log.go:32  [nsql2] 1699 ={"Timestamp":1640082482905304471,"EventType":"runtime_stats","MemRSSBytes":1038708736,"GoroutineCount":488,"MemStackSysBytes":4882432,"GoAllocBytes":67944096,"GoTotalBytes":151862312,"GoStatsStaleness":0.001672067,"HeapFragmentBytes":22438240,"HeapReservedBytes":40878080,"HeapReleasedBytes":14852096,"CGoAllocBytes":10247520,"CGoTotalBytes":18874368,"CGoCallRate":0.2999975,"CPUUserPercent":73.89938,"CPUSysPercent":5.299956,"GCPausePercent":0.005792692,"GCRunCount":125,"NetHostRecvBytes":93855,"NetHostSendBytes":93855}
I211221 10:28:02.906346 20821 2@server/status/runtime_log.go:47  [nsql2] 1700  runtime stats: 991 MiB RSS, 488 goroutines (stacks: 4.7 MiB), 65 MiB/145 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 39 MiB, heap released: 14 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 73.9/5.3 %(u/s)time, 0.0 %gc (125x), 92 KiB/92 KiB (r/w)net
I211221 10:28:03.112752 22808 jobs/registry.go:1179  [nsql2] 1701  AUTO CREATE STATS job 720935593620701186: stepping through state running with error: <nil>
I211221 10:28:03.254515 22808 jobs/registry.go:1179  [nsql2] 1702  AUTO CREATE STATS job 720935593620701186: stepping through state succeeded with error: <nil>
I211221 10:28:03.467663 21131 2@util/log/event_log.go:32  [nsql3] 1703 ={"Timestamp":1640082483467640714,"EventType":"runtime_stats","MemRSSBytes":988139520,"GoroutineCount":487,"MemStackSysBytes":5177344,"GoAllocBytes":62604704,"GoTotalBytes":153041960,"GoStatsStaleness":0.001902436,"HeapFragmentBytes":22157920,"HeapReservedBytes":47382528,"HeapReleasedBytes":13672448,"CGoAllocBytes":10247520,"CGoTotalBytes":18874368,"CGoCallRate":0.2999843,"CPUUserPercent":76.296005,"CPUSysPercent":5.4997125,"GCPausePercent":0.006175667,"GCRunCount":126,"NetHostRecvBytes":166178,"NetHostSendBytes":166178}
I211221 10:28:03.469064 21131 2@server/status/runtime_log.go:47  [nsql3] 1704  runtime stats: 942 MiB RSS, 487 goroutines (stacks: 4.9 MiB), 60 MiB/146 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 45 MiB, heap released: 13 MiB), 9.8 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 76.3/5.5 %(u/s)time, 0.0 %gc (126x), 162 KiB/162 KiB (r/w)net
I211221 10:28:03.657113 20941 jobs/adopt.go:106  [nsql2] 1705  claimed 1 jobs
I211221 10:28:03.678905 22801 jobs/adopt.go:247  [nsql2] 1706  job 720935395391275009: resuming execution
I211221 10:28:03.727969 22907 jobs/registry.go:1179  [nsql2] 1707  MIGRATION job 720935395391275009: stepping through state running with error: <nil>
I211221 10:28:03.802235 22907 jobs/registry.go:1179  [nsql2] 1708  MIGRATION job 720935395391275009: stepping through state succeeded with error: <nil>
I211221 10:28:04.688695 20943 server/settingswatcher/settings_watcher.go:112  [nsql2,rangefeed=settings] 1709  set cluster version to: 21.2-34
I211221 10:28:04.688793 21193 server/settingswatcher/settings_watcher.go:112  [nsql3,rangefeed=settings] 1710  set cluster version to: 21.2-34
I211221 10:28:04.752942 21213 util/log/event_log.go:32  [nsql3,client=127.0.0.1:53294,hostssl,user=root] 1711 ={"Timestamp":1640082423594091652,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211221 10:28:04.859449 21095 2@rpc/context.go:743  [nsql3] 1712  closing [{n1: 127.0.0.1:39483 (default)} {n0: 127.0.0.1:39483 (default)}]
I211221 10:28:04.859773 20862 2@rpc/context.go:743  [nsql2] 1713  closing [{n1: 127.0.0.1:39483 (default)} {n0: 127.0.0.1:39483 (default)}]
W211221 10:28:04.860674 21189 jobs/registry.go:851  [nsql3] 1714  canceling all adopted jobs due to stopper quiescing
I211221 10:28:04.860854 20788 2@rpc/context.go:743  [nsql2] 1715  closing [{n0: 127.0.0.1:39483 (system)}]
I211221 10:28:04.860679 21069 2@rpc/context.go:743  [nsql3] 1716  closing [{n0: 127.0.0.1:39483 (system)}]
W211221 10:28:04.865127 20822 ccl/kvccl/kvtenantccl/connector.go:182  [nsql2,tenant-connector] 1717  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:39544->127.0.0.1:39483: use of closed network connection
W211221 10:28:04.865674 21132 ccl/kvccl/kvtenantccl/connector.go:182  [nsql3,tenant-connector] 1718  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:39550->127.0.0.1:39483: use of closed network connection
W211221 10:28:04.867388 21141 sql/sqlliveness/slinstance/slinstance.go:222  [nsql3] 1719  exiting heartbeat loop
W211221 10:28:04.867512 20857 sql/sqlliveness/slinstance/slinstance.go:222  [nsql2] 1720  exiting heartbeat loop
W211221 10:28:04.870867 20939 kv/txn.go:712  [nsql2] 1721  failure aborting transaction: node unavailable; try another peer; abort caused by: could not query jobs table: cancel/pause-requested: context canceled
E211221 10:28:04.871580 20939 jobs/registry.go:809  [nsql2] 1722  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
W211221 10:28:04.872011 20939 jobs/registry.go:851  [nsql2] 1723  canceling all adopted jobs due to stopper quiescing
    testing.go:1152: race detected during execution of test
I211221 10:28:04.875047 14967 testutils/testcluster/testcluster.go:106  [-] 1724  TestCluster quiescing nodes
W211221 10:28:04.876335 15149 sql/sqlliveness/slinstance/slinstance.go:222  [n1] 1725  exiting heartbeat loop
I211221 10:28:04.881646 15045 2@rpc/context.go:743  [n1] 1726  closing [{n1: 127.0.0.1:39483 (default)}]
I211221 10:28:04.882661 15045 2@rpc/context.go:743  [n1] 1727  closing [{n0: 127.0.0.1:39483 (default)}]
W211221 10:28:04.882988 15104 jobs/registry.go:851  [n1] 1728  canceling all adopted jobs due to stopper quiescing
I211221 10:28:04.885060 16166 2@rpc/context.go:743  [n1] 1729  closing [{n1: 127.0.0.1:39483 (default)} {n0: 127.0.0.1:39483 (default)}]
I211221 10:28:04.893373 15067 kv/kvclient/kvcoord/transport_race.go:107  [-] 1730  transport race promotion: ran 216 iterations on up to 2994 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (70.96s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3985501&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3985501&tab=artifacts#/) on master @ [964695b941eba220e7830da13e5611b7e3ed6ac6](https://github.com/cockroachdb/cockroach/commits/964695b941eba220e7830da13e5611b7e3ed6ac6):


```
I211223 11:50:20.312675 19212 2@server/status/runtime_log.go:47  [n1] 1989  runtime stats: 900 MiB RSS, 489 goroutines (stacks: 5.3 MiB), 76 MiB/129 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 18 MiB, heap released: 3.5 MiB), 5.5 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 71.7/4.8 %(u/s)time, 0.0 %gc (177x), 107 KiB/107 KiB (r/w)net
I211223 11:50:20.410011 24854 2@util/log/event_log.go:32  [nsql2] 1990 ={"Timestamp":1640260220409997226,"EventType":"runtime_stats","MemRSSBytes":943886336,"GoroutineCount":491,"MemStackSysBytes":5570560,"GoAllocBytes":83748024,"GoTotalBytes":135622680,"GoStatsStaleness":0.00194826,"HeapFragmentBytes":17365832,"HeapReservedBytes":15482880,"HeapReleasedBytes":3661824,"CGoAllocBytes":5800032,"CGoTotalBytes":13946880,"CGoCallRate":0.29999104,"CPUUserPercent":72.197845,"CPUSysPercent":4.7998567,"GCPausePercent":0.007422068,"GCRunCount":177,"NetHostRecvBytes":107871,"NetHostSendBytes":107871}
I211223 11:50:20.411203 24854 2@server/status/runtime_log.go:47  [nsql2] 1991  runtime stats: 900 MiB RSS, 491 goroutines (stacks: 5.3 MiB), 80 MiB/129 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 15 MiB, heap released: 3.5 MiB), 5.5 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 72.2/4.8 %(u/s)time, 0.0 %gc (177x), 105 KiB/105 KiB (r/w)net
I211223 11:50:20.953020 25102 2@util/log/event_log.go:32  [nsql3] 1992 ={"Timestamp":1640260220953006565,"EventType":"runtime_stats","MemRSSBytes":939679744,"GoroutineCount":489,"MemStackSysBytes":5242880,"GoAllocBytes":51727200,"GoTotalBytes":136048664,"GoStatsStaleness":0.001855176,"HeapFragmentBytes":20550816,"HeapReservedBytes":45072384,"HeapReleasedBytes":3235840,"CGoAllocBytes":5800032,"CGoTotalBytes":13946880,"CGoCallRate":0.30000216,"CPUUserPercent":70.50051,"CPUSysPercent":4.9000354,"GCPausePercent":0.007520464,"GCRunCount":178,"NetHostRecvBytes":107991,"NetHostSendBytes":107991}
I211223 11:50:20.954021 25102 2@server/status/runtime_log.go:47  [nsql3] 1993  runtime stats: 896 MiB RSS, 489 goroutines (stacks: 5.0 MiB), 49 MiB/130 MiB Go alloc/total(stale) (heap fragmentation: 20 MiB, heap reserved: 43 MiB, heap released: 3.1 MiB), 5.5 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 70.5/4.9 %(u/s)time, 0.0 %gc (178x), 106 KiB/106 KiB (r/w)net
I211223 11:50:21.667789 19189 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1994  [JOB 6] WAL created 000007
I211223 11:50:30.305595 19207 2@gossip/gossip.go:566  [n1] 1995  gossip status (ok, 1 node)
I211223 11:50:30.305595 19207 2@gossip/gossip.go:566  [n1] 1995 +gossip client (0/3 cur/max conns)
I211223 11:50:30.305595 19207 2@gossip/gossip.go:566  [n1] 1995 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211223 11:50:30.305595 19207 2@gossip/gossip.go:566  [n1] 1995 +gossip connectivity
I211223 11:50:30.305595 19207 2@gossip/gossip.go:566  [n1] 1995 +  n1 [sentinel];
I211223 11:50:30.312115 19212 2@util/log/event_log.go:32  [n1] 1996 ={"Timestamp":1640260230312104494,"EventType":"runtime_stats","MemRSSBytes":961916928,"GoroutineCount":489,"MemStackSysBytes":5799936,"GoAllocBytes":90166864,"GoTotalBytes":136876056,"GoStatsStaleness":0.001857765,"HeapFragmentBytes":17615280,"HeapReservedBytes":9838592,"HeapReleasedBytes":2408448,"CGoAllocBytes":9994464,"CGoTotalBytes":18145280,"CGoCallRate":0.3999834,"CPUUserPercent":62.89739,"CPUSysPercent":4.4998136,"GCPausePercent":0.0052497922,"GCRunCount":181,"NetHostRecvBytes":92515,"NetHostSendBytes":92515}
I211223 11:50:30.313154 19212 2@server/status/runtime_log.go:47  [n1] 1997  runtime stats: 917 MiB RSS, 489 goroutines (stacks: 5.5 MiB), 86 MiB/130 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 9.4 MiB, heap released: 2.3 MiB), 9.5 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 62.9/4.5 %(u/s)time, 0.0 %gc (181x), 90 KiB/90 KiB (r/w)net
I211223 11:50:30.413873 24854 2@util/log/event_log.go:32  [nsql2] 1998 ={"Timestamp":1640260230413857044,"EventType":"runtime_stats","MemRSSBytes":964530176,"GoroutineCount":488,"MemStackSysBytes":4751360,"GoAllocBytes":92927216,"GoTotalBytes":136876056,"GoStatsStaleness":0.002549276,"HeapFragmentBytes":16304912,"HeapReservedBytes":9338880,"HeapReleasedBytes":2408448,"CGoAllocBytes":9994464,"CGoTotalBytes":18145280,"CGoCallRate":0.3998457,"CPUUserPercent":63.575466,"CPUSysPercent":4.598225,"GCPausePercent":0.0076577445,"GCRunCount":182,"NetHostRecvBytes":92515,"NetHostSendBytes":92515}
I211223 11:50:30.415638 24854 2@server/status/runtime_log.go:47  [nsql2] 1999  runtime stats: 920 MiB RSS, 488 goroutines (stacks: 4.5 MiB), 89 MiB/130 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 8.9 MiB, heap released: 2.3 MiB), 9.5 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 63.6/4.6 %(u/s)time, 0.0 %gc (182x), 90 KiB/90 KiB (r/w)net
I211223 11:50:30.954323 25102 2@util/log/event_log.go:32  [nsql3] 2000 ={"Timestamp":1640260230954306118,"EventType":"runtime_stats","MemRSSBytes":965619712,"GoroutineCount":490,"MemStackSysBytes":5996544,"GoAllocBytes":68179944,"GoTotalBytes":136876056,"GoStatsStaleness":0.001973593,"HeapFragmentBytes":16525336,"HeapReservedBytes":32718848,"HeapReleasedBytes":2408448,"CGoAllocBytes":9994464,"CGoTotalBytes":18145280,"CGoCallRate":0.39994806,"CPUUserPercent":67.49124,"CPUSysPercent":4.899364,"GCPausePercent":0.005566867,"GCRunCount":182,"NetHostRecvBytes":90227,"NetHostSendBytes":90227}
I211223 11:50:30.955603 25102 2@server/status/runtime_log.go:47  [nsql3] 2001  runtime stats: 921 MiB RSS, 490 goroutines (stacks: 5.7 MiB), 65 MiB/130 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 31 MiB, heap released: 2.3 MiB), 9.5 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 67.5/4.9 %(u/s)time, 0.0 %gc (182x), 88 KiB/88 KiB (r/w)net
I211223 11:50:33.553527 20150 server/auto_upgrade.go:44  [n1] 2002  auto upgrade disabled by testing
I211223 11:50:40.312259 19212 2@util/log/event_log.go:32  [n1] 2003 ={"Timestamp":1640260240312245703,"EventType":"runtime_stats","MemRSSBytes":978903040,"GoroutineCount":488,"MemStackSysBytes":5111808,"GoAllocBytes":72565056,"GoTotalBytes":140750872,"GoStatsStaleness":0.001989013,"HeapFragmentBytes":18759360,"HeapReservedBytes":30859264,"HeapReleasedBytes":2727936,"CGoAllocBytes":9994464,"CGoTotalBytes":18145280,"CGoCallRate":0.29999578,"CPUUserPercent":70.59901,"CPUSysPercent":5.1999273,"GCPausePercent":0.010390635,"GCRunCount":186,"NetHostRecvBytes":94583,"NetHostSendBytes":94583}
I211223 11:50:40.313415 19212 2@server/status/runtime_log.go:47  [n1] 2004  runtime stats: 934 MiB RSS, 488 goroutines (stacks: 4.9 MiB), 69 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 29 MiB, heap released: 2.6 MiB), 9.5 MiB/17 MiB CGO alloc/total (0.3 CGO/sec), 70.6/5.2 %(u/s)time, 0.0 %gc (186x), 92 KiB/92 KiB (r/w)net
I211223 11:50:40.413256 24854 2@util/log/event_log.go:32  [nsql2] 2005 ={"Timestamp":1640260240413241329,"EventType":"runtime_stats","MemRSSBytes":978825216,"GoroutineCount":488,"MemStackSysBytes":5177344,"GoAllocBytes":74321432,"GoTotalBytes":140750872,"GoStatsStaleness":0.00190088,"HeapFragmentBytes":18428392,"HeapReservedBytes":29368320,"HeapReleasedBytes":2727936,"CGoAllocBytes":9994464,"CGoTotalBytes":18145280,"CGoCallRate":0.30001843,"CPUUserPercent":69.10424,"CPUSysPercent":5.000307,"GCPausePercent":0.00798058,"GCRunCount":186,"NetHostRecvBytes":95243,"NetHostSendBytes":95243}
I211223 11:50:40.414365 24854 2@server/status/runtime_log.go:47  [nsql2] 2006  runtime stats: 934 MiB RSS, 488 goroutines (stacks: 4.9 MiB), 71 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 28 MiB, heap released: 2.6 MiB), 9.5 MiB/17 MiB CGO alloc/total (0.3 CGO/sec), 69.1/5.0 %(u/s)time, 0.0 %gc (186x), 93 KiB/93 KiB (r/w)net
I211223 11:50:40.570644 25133 jobs/adopt.go:106  [nsql3] 2007  claimed 1 jobs
I211223 11:50:40.598109 26778 jobs/adopt.go:247  [nsql3] 2008  job 721517871215869953: resuming execution
I211223 11:50:40.632073 26767 jobs/registry.go:1187  [nsql2] 2009  AUTO CREATE STATS job 721518069427666946: stepping through state running with error: <nil>
I211223 11:50:40.669360 26821 jobs/registry.go:1187  [nsql3] 2010  MIGRATION job 721517871215869953: stepping through state running with error: <nil>
I211223 11:50:40.715542 26821 jobs/registry.go:1187  [nsql3] 2011  MIGRATION job 721517871215869953: stepping through state succeeded with error: <nil>
I211223 11:50:40.836701 26767 jobs/registry.go:1187  [nsql2] 2012  AUTO CREATE STATS job 721518069427666946: stepping through state succeeded with error: <nil>
I211223 11:50:40.836974 25135 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 2013  set cluster version to: 21.2-34
I211223 11:50:40.837753 24946 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 2014  set cluster version to: 21.2-34
I211223 11:50:40.910318 25266 util/log/event_log.go:32  [nsql3,client=127.0.0.1:53600,hostssl,user=root] 2015 ={"Timestamp":1640260181084871378,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211223 11:50:40.955051 25102 2@util/log/event_log.go:32  [nsql3] 2016 ={"Timestamp":1640260240955038231,"EventType":"runtime_stats","MemRSSBytes":918003712,"GoroutineCount":488,"MemStackSysBytes":5865472,"GoAllocBytes":97343296,"GoTotalBytes":142725144,"GoStatsStaleness":0.002172568,"HeapFragmentBytes":13953216,"HeapReservedBytes":11845632,"HeapReleasedBytes":1015808,"CGoAllocBytes":9994464,"CGoTotalBytes":18145280,"CGoCallRate":0.299978,"CPUUserPercent":79.49416,"CPUSysPercent":6.7995014,"GCPausePercent":0.010585453,"GCRunCount":187,"NetHostRecvBytes":224979,"NetHostSendBytes":224979}
I211223 11:50:40.956635 25102 2@server/status/runtime_log.go:47  [nsql3] 2017  runtime stats: 876 MiB RSS, 488 goroutines (stacks: 5.6 MiB), 93 MiB/136 MiB Go alloc/total(stale) (heap fragmentation: 13 MiB, heap reserved: 11 MiB, heap released: 992 KiB), 9.5 MiB/17 MiB CGO alloc/total (0.3 CGO/sec), 79.5/6.8 %(u/s)time, 0.0 %gc (187x), 220 KiB/220 KiB (r/w)net
I211223 11:50:41.093782 25139 2@rpc/context.go:743  [nsql3] 2018  closing [{n1: 127.0.0.1:34175 (default)} {n0: 127.0.0.1:34175 (default)}]
I211223 11:50:41.094710 25107 2@rpc/context.go:743  [nsql3] 2019  closing [{n0: 127.0.0.1:34175 (system)}]
I211223 11:50:41.095137 24800 2@rpc/context.go:743  [nsql2] 2020  closing [{n0: 127.0.0.1:34175 (system)}]
I211223 11:50:41.095400 24889 2@rpc/context.go:743  [nsql2] 2021  closing [{n1: 127.0.0.1:34175 (default)} {n0: 127.0.0.1:34175 (default)}]
W211223 11:50:41.096399 24910 jobs/registry.go:859  [nsql2] 2022  canceling all adopted jobs due to stopper quiescing
W211223 11:50:41.097061 25131 jobs/registry.go:859  [nsql3] 2023  canceling all adopted jobs due to stopper quiescing
W211223 11:50:41.097591 24977 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 2024  exiting heartbeat loop
W211223 11:50:41.098844 24855 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 2025  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:46084->127.0.0.1:34175: use of closed network connection
W211223 11:50:41.099701 25103 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 2026  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: EOF
W211223 11:50:41.100134 24883 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 2027  exiting heartbeat loop
    testing.go:1152: race detected during execution of test
I211223 11:50:41.109379 17487 testutils/testcluster/testcluster.go:106  [-] 2028  TestCluster quiescing nodes
I211223 11:50:41.110198 20012 2@rpc/context.go:743  [n1] 2029  closing [{n1: 127.0.0.1:34175 (default)} {n0: 127.0.0.1:34175 (default)}]
W211223 11:50:41.111736 19025 jobs/registry.go:859  [n1] 2030  canceling all adopted jobs due to stopper quiescing
W211223 11:50:41.113810 19238 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 2031  exiting heartbeat loop
I211223 11:50:41.114369 18927 kv/kvclient/kvcoord/transport_race.go:107  [-] 2032  transport race promotion: ran 213 iterations on up to 2899 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (69.78s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3993962&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3993962&tab=artifacts#/) on master @ [8c1528fff5c132b971428bc7f18bf0ea5eb16799](https://github.com/cockroachdb/cockroach/commits/8c1528fff5c132b971428bc7f18bf0ea5eb16799):


```
I211224 11:35:33.146987 23590 2@server/status/runtime_log.go:47  [nsql3] 1663  runtime stats: 1.2 GiB RSS, 480 goroutines (stacks: 5.5 MiB), 142 MiB/198 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 12 MiB, heap released: 1.5 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 75.2/4.6 %(u/s)time, 0.0 %gc (132x), 98 KiB/98 KiB (r/w)net
I211224 11:35:33.240395 17718 2@util/log/event_log.go:32  [n1] 1664 ={"Timestamp":1640345733240383910,"EventType":"runtime_stats","MemRSSBytes":1282011136,"GoroutineCount":481,"MemStackSysBytes":5767168,"GoAllocBytes":149291080,"GoTotalBytes":208002088,"GoStatsStaleness":0.001927365,"HeapFragmentBytes":23633848,"HeapReservedBytes":12673024,"HeapReleasedBytes":1572864,"CGoAllocBytes":5995744,"CGoTotalBytes":13230080,"CGoCallRate":0.2999966,"CPUUserPercent":74.299164,"CPUSysPercent":4.5999484,"GCPausePercent":0.0038933363,"GCRunCount":132,"NetHostRecvBytes":100028,"NetHostSendBytes":100028}
I211224 11:35:33.241438 17718 2@server/status/runtime_log.go:47  [n1] 1665  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 5.5 MiB), 142 MiB/198 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 12 MiB, heap released: 1.5 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 74.3/4.6 %(u/s)time, 0.0 %gc (132x), 98 KiB/98 KiB (r/w)net
I211224 11:35:41.234953 23221 sql/sqlliveness/slstorage/slstorage.go:388  [nsql2] 1666  deleted 1 expired SQL liveness sessions
I211224 11:35:42.653081 23307 2@util/log/event_log.go:32  [nsql2] 1667 ={"Timestamp":1640345742653064744,"EventType":"runtime_stats","MemRSSBytes":1279623168,"GoroutineCount":481,"MemStackSysBytes":5570560,"GoAllocBytes":103265528,"GoTotalBytes":208002088,"GoStatsStaleness":0.001934002,"HeapFragmentBytes":27052808,"HeapReservedBytes":55476224,"HeapReleasedBytes":1572864,"CGoAllocBytes":5995744,"CGoTotalBytes":13230080,"CGoCallRate":0.29999462,"CPUUserPercent":66.998795,"CPUSysPercent":4.799914,"GCPausePercent":0.019689657,"GCRunCount":135,"NetHostRecvBytes":106959,"NetHostSendBytes":106959}
I211224 11:35:42.654209 23307 2@server/status/runtime_log.go:47  [nsql2] 1668  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 5.3 MiB), 98 MiB/198 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 53 MiB, heap released: 1.5 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 67.0/4.8 %(u/s)time, 0.0 %gc (135x), 104 KiB/104 KiB (r/w)net
I211224 11:35:43.146486 23590 2@util/log/event_log.go:32  [nsql3] 1669 ={"Timestamp":1640345743146470821,"EventType":"runtime_stats","MemRSSBytes":1275310080,"GoroutineCount":481,"MemStackSysBytes":5865472,"GoAllocBytes":116901432,"GoTotalBytes":208002088,"GoStatsStaleness":0.001849339,"HeapFragmentBytes":24852936,"HeapReservedBytes":43745280,"HeapReleasedBytes":1572864,"CGoAllocBytes":5995744,"CGoTotalBytes":13230080,"CGoCallRate":0.29997757,"CPUUserPercent":67.99492,"CPUSysPercent":4.6996484,"GCPausePercent":0.019688537,"GCRunCount":135,"NetHostRecvBytes":112654,"NetHostSendBytes":112654}
I211224 11:35:43.147621 23590 2@server/status/runtime_log.go:47  [nsql3] 1670  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 5.6 MiB), 112 MiB/198 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 42 MiB, heap released: 1.5 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 68.0/4.7 %(u/s)time, 0.0 %gc (135x), 110 KiB/110 KiB (r/w)net
I211224 11:35:43.241194 17718 2@util/log/event_log.go:32  [n1] 1671 ={"Timestamp":1640345743241181836,"EventType":"runtime_stats","MemRSSBytes":1275428864,"GoroutineCount":481,"MemStackSysBytes":5865472,"GoAllocBytes":117126648,"GoTotalBytes":208002088,"GoStatsStaleness":0.001868992,"HeapFragmentBytes":24701448,"HeapReservedBytes":43671552,"HeapReleasedBytes":1572864,"CGoAllocBytes":5995744,"CGoTotalBytes":13230080,"CGoCallRate":0.29997608,"CPUUserPercent":67.994576,"CPUSysPercent":4.599633,"GCPausePercent":0.01968844,"GCRunCount":135,"NetHostRecvBytes":111485,"NetHostSendBytes":111485}
I211224 11:35:43.242271 17718 2@server/status/runtime_log.go:47  [n1] 1672  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 5.6 MiB), 112 MiB/198 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 42 MiB, heap released: 1.5 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 68.0/4.6 %(u/s)time, 0.0 %gc (135x), 109 KiB/109 KiB (r/w)net
I211224 11:35:44.564352 17642 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1673  [JOB 6] WAL created 000007
I211224 11:35:52.653877 23307 2@util/log/event_log.go:32  [nsql2] 1674 ={"Timestamp":1640345752653863086,"EventType":"runtime_stats","MemRSSBytes":1317994496,"GoroutineCount":481,"MemStackSysBytes":4947968,"GoAllocBytes":92181688,"GoTotalBytes":210541608,"GoStatsStaleness":0.001795196,"HeapFragmentBytes":27896648,"HeapReservedBytes":68878336,"HeapReleasedBytes":3227648,"CGoAllocBytes":10190176,"CGoTotalBytes":17424384,"CGoCallRate":0.39996806,"CPUUserPercent":73.794106,"CPUSysPercent":5.199585,"GCPausePercent":0.002903948,"GCRunCount":138,"NetHostRecvBytes":100111,"NetHostSendBytes":100111}
I211224 11:35:52.654929 23307 2@server/status/runtime_log.go:47  [nsql2] 1675  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 4.7 MiB), 88 MiB/201 MiB Go alloc/total(stale) (heap fragmentation: 27 MiB, heap reserved: 66 MiB, heap released: 3.1 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 73.8/5.2 %(u/s)time, 0.0 %gc (138x), 98 KiB/98 KiB (r/w)net
I211224 11:35:53.147136 23590 2@util/log/event_log.go:32  [nsql3] 1676 ={"Timestamp":1640345753147120797,"EventType":"runtime_stats","MemRSSBytes":1311948800,"GoroutineCount":481,"MemStackSysBytes":5144576,"GoAllocBytes":99112920,"GoTotalBytes":210541608,"GoStatsStaleness":0.002033326,"HeapFragmentBytes":26880040,"HeapReservedBytes":62767104,"HeapReleasedBytes":3227648,"CGoAllocBytes":10190176,"CGoTotalBytes":17424384,"CGoCallRate":0.39997402,"CPUUserPercent":71.59535,"CPUSysPercent":5.199662,"GCPausePercent":0.0029039914,"GCRunCount":138,"NetHostRecvBytes":91935,"NetHostSendBytes":91935}
I211224 11:35:53.148219 23590 2@server/status/runtime_log.go:47  [nsql3] 1677  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 4.9 MiB), 94 MiB/201 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 60 MiB, heap released: 3.1 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 71.6/5.2 %(u/s)time, 0.0 %gc (138x), 90 KiB/90 KiB (r/w)net
I211224 11:35:53.233602 17569 2@gossip/gossip.go:566  [n1] 1678  gossip status (ok, 1 node)
I211224 11:35:53.233602 17569 2@gossip/gossip.go:566  [n1] 1678 +gossip client (0/3 cur/max conns)
I211224 11:35:53.233602 17569 2@gossip/gossip.go:566  [n1] 1678 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211224 11:35:53.233602 17569 2@gossip/gossip.go:566  [n1] 1678 +gossip connectivity
I211224 11:35:53.233602 17569 2@gossip/gossip.go:566  [n1] 1678 +  n1 [sentinel];
I211224 11:35:53.241934 17718 2@util/log/event_log.go:32  [n1] 1679 ={"Timestamp":1640345753241923707,"EventType":"runtime_stats","MemRSSBytes":1310920704,"GoroutineCount":481,"MemStackSysBytes":5308416,"GoAllocBytes":103137576,"GoTotalBytes":210541608,"GoStatsStaleness":0.001919007,"HeapFragmentBytes":26418904,"HeapReservedBytes":59039744,"HeapReleasedBytes":3227648,"CGoAllocBytes":10190176,"CGoTotalBytes":17424384,"CGoCallRate":0.3999703,"CPUUserPercent":72.594604,"CPUSysPercent":5.2996063,"GCPausePercent":0.0029039644,"GCRunCount":138,"NetHostRecvBytes":92851,"NetHostSendBytes":92851}
I211224 11:35:53.243056 17718 2@server/status/runtime_log.go:47  [n1] 1680  runtime stats: 1.2 GiB RSS, 481 goroutines (stacks: 5.1 MiB), 98 MiB/201 MiB Go alloc/total(stale) (heap fragmentation: 25 MiB, heap reserved: 56 MiB, heap released: 3.1 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 72.6/5.3 %(u/s)time, 0.0 %gc (138x), 91 KiB/91 KiB (r/w)net
I211224 11:35:57.048177 18600 server/auto_upgrade.go:44  [n1] 1681  auto upgrade disabled by testing
I211224 11:36:01.229002 23400 jobs/adopt.go:106  [nsql2] 1682  claimed 1 jobs
I211224 11:36:01.248148 25068 jobs/adopt.go:247  [nsql2] 1683  job 721798110516903937: resuming execution
I211224 11:36:01.300604 25258 jobs/registry.go:1187  [nsql2] 1684  MIGRATION job 721798110516903937: stepping through state running with error: <nil>
I211224 11:36:01.369373 25258 jobs/registry.go:1187  [nsql2] 1685  MIGRATION job 721798110516903937: stepping through state succeeded with error: <nil>
I211224 11:36:01.617281 23402 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 1686  set cluster version to: 21.2-34
I211224 11:36:01.617441 23656 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 1687  set cluster version to: 21.2-34
I211224 11:36:01.666887 23649 util/log/event_log.go:32  [nsql3,client=127.0.0.1:44208,hostssl,user=root] 1688 ={"Timestamp":1640345703280692221,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211224 11:36:01.811796 23604 2@rpc/context.go:743  [nsql3] 1689  closing [{n1: 127.0.0.1:36183 (default)} {n0: 127.0.0.1:36183 (default)}]
I211224 11:36:01.812735 23574 2@rpc/context.go:743  [nsql3] 1690  closing [{n0: 127.0.0.1:36183 (system)}]
I211224 11:36:01.812758 23480 2@rpc/context.go:743  [nsql3] 1691  closing [{n0: 127.0.0.1:36183 (system)}]
I211224 11:36:01.812956 23255 2@rpc/context.go:743  [nsql2] 1692  closing [{n1: 127.0.0.1:36183 (default)} {n0: 127.0.0.1:36183 (default)}]
W211224 11:36:01.813658 23398 jobs/registry.go:859  [nsql2] 1694  canceling all adopted jobs due to stopper quiescing
W211224 11:36:01.813532 23591 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 1693  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211224 11:36:01.813733 23652 jobs/registry.go:859  [nsql3] 1695  canceling all adopted jobs due to stopper quiescing
W211224 11:36:01.813991 23531 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 1696  exiting heartbeat loop
I211224 11:36:01.813658 23245 2@rpc/context.go:743  [nsql2] 1697  closing [{n0: 127.0.0.1:36183 (system)}]
W211224 11:36:01.816713 23222 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 1698  exiting heartbeat loop
W211224 11:36:01.816935 23308 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 1699  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:40008->127.0.0.1:36183: use of closed network connection
    testing.go:1152: race detected during execution of test
I211224 11:36:01.824823 17424 testutils/testcluster/testcluster.go:106  [-] 1700  TestCluster quiescing nodes
I211224 11:36:01.829112 17532 2@rpc/context.go:743  [n1] 1701  closing [{n1: 127.0.0.1:36183 (default)}]
W211224 11:36:01.829853 17709 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 1702  exiting heartbeat loop
W211224 11:36:01.830465 17743 jobs/registry.go:859  [n1] 1703  canceling all adopted jobs due to stopper quiescing
I211224 11:36:01.830693 17532 2@rpc/context.go:743  [n1] 1704  closing [{n0: 127.0.0.1:36183 (default)}]
I211224 11:36:01.831072 18646 2@rpc/context.go:743  [n1] 1705  closing [{n1: 127.0.0.1:36183 (default)} {n0: 127.0.0.1:36183 (default)}]
I211224 11:36:02.046152 17439 kv/kvclient/kvcoord/transport_race.go:107  [-] 1706  transport race promotion: ran 230 iterations on up to 2968 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (67.60s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3999480&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3999480&tab=artifacts#/) on master @ [f3710ae65e327d0795123029db679f8c9f3e610e](https://github.com/cockroachdb/cockroach/commits/f3710ae65e327d0795123029db679f8c9f3e610e):


```
I211228 11:07:16.002611 16868 2@util/log/event_log.go:32  [n1] 1676 ={"Timestamp":1640689636002599285,"EventType":"runtime_stats","MemRSSBytes":902778880,"GoroutineCount":481,"MemStackSysBytes":5242880,"GoAllocBytes":77678112,"GoTotalBytes":136777752,"GoStatsStaleness":0.001986143,"HeapFragmentBytes":17512928,"HeapReservedBytes":22626304,"HeapReleasedBytes":2768896,"CGoAllocBytes":5862368,"CGoTotalBytes":14815232,"CGoCallRate":0.29999173,"CPUUserPercent":68.59811,"CPUSysPercent":3.8998923,"GCPausePercent":0.0077254074,"GCRunCount":163,"NetHostRecvBytes":97972,"NetHostSendBytes":97972}
I211228 11:07:16.003794 16868 2@server/status/runtime_log.go:47  [n1] 1677  runtime stats: 861 MiB RSS, 481 goroutines (stacks: 5.0 MiB), 74 MiB/130 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 22 MiB, heap released: 2.6 MiB), 5.6 MiB/14 MiB CGO alloc/total (0.3 CGO/sec), 68.6/3.9 %(u/s)time, 0.0 %gc (163x), 96 KiB/96 KiB (r/w)net
I211228 11:07:16.465259 22430 sql/sqlliveness/slstorage/slstorage.go:388  [nsql2] 1678  deleted 1 expired SQL liveness sessions
I211228 11:07:16.517832 22765 2@util/log/event_log.go:32  [nsql3] 1679 ={"Timestamp":1640689636517815838,"EventType":"runtime_stats","MemRSSBytes":891928576,"GoroutineCount":481,"MemStackSysBytes":5013504,"GoAllocBytes":54945384,"GoTotalBytes":136777752,"GoStatsStaleness":0.001874616,"HeapFragmentBytes":19052952,"HeapReservedBytes":44048384,"HeapReleasedBytes":2768896,"CGoAllocBytes":5862368,"CGoTotalBytes":14815232,"CGoCallRate":0.29998493,"CPUUserPercent":70.39646,"CPUSysPercent":4.099794,"GCPausePercent":0.0086812135,"GCRunCount":164,"NetHostRecvBytes":103472,"NetHostSendBytes":103472}
I211228 11:07:16.518984 22765 2@server/status/runtime_log.go:47  [nsql3] 1680  runtime stats: 851 MiB RSS, 481 goroutines (stacks: 4.8 MiB), 52 MiB/130 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 42 MiB, heap released: 2.6 MiB), 5.6 MiB/14 MiB CGO alloc/total (0.3 CGO/sec), 70.4/4.1 %(u/s)time, 0.0 %gc (164x), 101 KiB/101 KiB (r/w)net
I211228 11:07:17.340065 16813 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1681  [JOB 6] WAL created 000007
I211228 11:07:25.959809 22464 2@util/log/event_log.go:32  [nsql2] 1682 ={"Timestamp":1640689645959794016,"EventType":"runtime_stats","MemRSSBytes":913227776,"GoroutineCount":481,"MemStackSysBytes":5439488,"GoAllocBytes":86394272,"GoTotalBytes":138391576,"GoStatsStaleness":0.001825662,"HeapFragmentBytes":17332832,"HeapReservedBytes":15507456,"HeapReleasedBytes":5349376,"CGoAllocBytes":10056800,"CGoTotalBytes":19013632,"CGoCallRate":0.39997318,"CPUUserPercent":63.295757,"CPUSysPercent":4.8996716,"GCPausePercent":0.00625788,"GCRunCount":167,"NetHostRecvBytes":104281,"NetHostSendBytes":104281}
I211228 11:07:25.960976 22464 2@server/status/runtime_log.go:47  [nsql2] 1683  runtime stats: 871 MiB RSS, 481 goroutines (stacks: 5.2 MiB), 82 MiB/132 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 15 MiB, heap released: 5.1 MiB), 9.6 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 63.3/4.9 %(u/s)time, 0.0 %gc (167x), 102 KiB/102 KiB (r/w)net
I211228 11:07:25.997984 16863 2@gossip/gossip.go:566  [n1] 1684  gossip status (ok, 1 node)
I211228 11:07:25.997984 16863 2@gossip/gossip.go:566  [n1] 1684 +gossip client (0/3 cur/max conns)
I211228 11:07:25.997984 16863 2@gossip/gossip.go:566  [n1] 1684 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211228 11:07:25.997984 16863 2@gossip/gossip.go:566  [n1] 1684 +gossip connectivity
I211228 11:07:25.997984 16863 2@gossip/gossip.go:566  [n1] 1684 +  n1 [sentinel];
I211228 11:07:26.003946 16868 2@util/log/event_log.go:32  [n1] 1685 ={"Timestamp":1640689646003936426,"EventType":"runtime_stats","MemRSSBytes":913367040,"GoroutineCount":481,"MemStackSysBytes":5505024,"GoAllocBytes":87846120,"GoTotalBytes":138391576,"GoStatsStaleness":0.00235685,"HeapFragmentBytes":16978712,"HeapReservedBytes":14344192,"HeapReleasedBytes":5349376,"CGoAllocBytes":10056800,"CGoTotalBytes":19013632,"CGoCallRate":0.3999465,"CPUUserPercent":63.691486,"CPUSysPercent":4.9993315,"GCPausePercent":0.0062574632,"GCRunCount":167,"NetHostRecvBytes":104281,"NetHostSendBytes":104281}
I211228 11:07:26.004951 16868 2@server/status/runtime_log.go:47  [n1] 1686  runtime stats: 871 MiB RSS, 481 goroutines (stacks: 5.3 MiB), 84 MiB/132 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 14 MiB, heap released: 5.1 MiB), 9.6 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 63.7/5.0 %(u/s)time, 0.0 %gc (167x), 102 KiB/102 KiB (r/w)net
I211228 11:07:26.518992 22765 2@util/log/event_log.go:32  [nsql3] 1687 ={"Timestamp":1640689646518976790,"EventType":"runtime_stats","MemRSSBytes":918556672,"GoroutineCount":482,"MemStackSysBytes":5734400,"GoAllocBytes":65508440,"GoTotalBytes":138391576,"GoStatsStaleness":0.001964578,"HeapFragmentBytes":16419752,"HeapReservedBytes":37011456,"HeapReleasedBytes":5349376,"CGoAllocBytes":10056800,"CGoTotalBytes":19013632,"CGoCallRate":0.39995357,"CPUUserPercent":67.492165,"CPUSysPercent":5.6993384,"GCPausePercent":0.0079145115,"GCRunCount":168,"NetHostRecvBytes":97340,"NetHostSendBytes":97340}
I211228 11:07:26.520158 22765 2@server/status/runtime_log.go:47  [nsql3] 1688  runtime stats: 876 MiB RSS, 482 goroutines (stacks: 5.5 MiB), 62 MiB/132 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 35 MiB, heap released: 5.1 MiB), 9.6 MiB/18 MiB CGO alloc/total (0.4 CGO/sec), 67.5/5.7 %(u/s)time, 0.0 %gc (168x), 95 KiB/95 KiB (r/w)net
I211228 11:07:29.695697 17792 server/auto_upgrade.go:44  [n1] 1689  auto upgrade disabled by testing
I211228 11:07:35.960924 22464 2@util/log/event_log.go:32  [nsql2] 1690 ={"Timestamp":1640689655960909274,"EventType":"runtime_stats","MemRSSBytes":930099200,"GoroutineCount":480,"MemStackSysBytes":4751360,"GoAllocBytes":71733600,"GoTotalBytes":140496920,"GoStatsStaleness":0.001802863,"HeapFragmentBytes":18009760,"HeapReservedBytes":32284672,"HeapReleasedBytes":3244032,"CGoAllocBytes":10056800,"CGoTotalBytes":19013632,"CGoCallRate":0.29996657,"CPUUserPercent":73.89176,"CPUSysPercent":6.0993204,"GCPausePercent":0.00942594,"GCRunCount":172,"NetHostRecvBytes":97816,"NetHostSendBytes":97816}
I211228 11:07:35.962069 22464 2@server/status/runtime_log.go:47  [nsql2] 1691  runtime stats: 887 MiB RSS, 480 goroutines (stacks: 4.5 MiB), 68 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 31 MiB, heap released: 3.1 MiB), 9.6 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 73.9/6.1 %(u/s)time, 0.0 %gc (172x), 96 KiB/96 KiB (r/w)net
I211228 11:07:36.003897 16868 2@util/log/event_log.go:32  [n1] 1692 ={"Timestamp":1640689656003885285,"EventType":"runtime_stats","MemRSSBytes":929394688,"GoroutineCount":480,"MemStackSysBytes":4816896,"GoAllocBytes":74121112,"GoTotalBytes":140496920,"GoStatsStaleness":0.001918755,"HeapFragmentBytes":17662056,"HeapReservedBytes":30179328,"HeapReleasedBytes":3244032,"CGoAllocBytes":10056800,"CGoTotalBytes":19013632,"CGoCallRate":0.30000153,"CPUUserPercent":74.000374,"CPUSysPercent":6.0000305,"GCPausePercent":0.009427038,"GCRunCount":172,"NetHostRecvBytes":98658,"NetHostSendBytes":98658}
I211228 11:07:36.005031 16868 2@server/status/runtime_log.go:47  [n1] 1693  runtime stats: 886 MiB RSS, 480 goroutines (stacks: 4.6 MiB), 71 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 29 MiB, heap released: 3.1 MiB), 9.6 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 74.0/6.0 %(u/s)time, 0.0 %gc (172x), 96 KiB/96 KiB (r/w)net
I211228 11:07:36.188440 24406 jobs/registry.go:1187  [nsql2] 1694  AUTO CREATE STATS job 722925178296139778: stepping through state running with error: <nil>
I211228 11:07:36.358649 24406 jobs/registry.go:1187  [nsql2] 1695  AUTO CREATE STATS job 722925178296139778: stepping through state succeeded with error: <nil>
I211228 11:07:36.519214 22765 2@util/log/event_log.go:32  [nsql3] 1696 ={"Timestamp":1640689656519198129,"EventType":"runtime_stats","MemRSSBytes":894840832,"GoroutineCount":479,"MemStackSysBytes":5111808,"GoAllocBytes":70725528,"GoTotalBytes":140496920,"GoStatsStaleness":0.001783101,"HeapFragmentBytes":16412776,"HeapReservedBytes":34529280,"HeapReleasedBytes":3244032,"CGoAllocBytes":10056800,"CGoTotalBytes":19013632,"CGoCallRate":0.2999934,"CPUUserPercent":75.89833,"CPUSysPercent":6.8998485,"GCPausePercent":0.008878985,"GCRunCount":173,"NetHostRecvBytes":170561,"NetHostSendBytes":170561}
I211228 11:07:36.520343 22765 2@server/status/runtime_log.go:47  [nsql3] 1697  runtime stats: 853 MiB RSS, 479 goroutines (stacks: 4.9 MiB), 67 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 33 MiB, heap released: 3.1 MiB), 9.6 MiB/18 MiB CGO alloc/total (0.3 CGO/sec), 75.9/6.9 %(u/s)time, 0.0 %gc (173x), 167 KiB/167 KiB (r/w)net
I211228 11:07:39.165849 22857 jobs/adopt.go:106  [nsql3] 1698  claimed 1 jobs
I211228 11:07:39.184754 24554 jobs/adopt.go:247  [nsql3] 1699  job 722924980096368641: resuming execution
I211228 11:07:39.238099 24556 jobs/registry.go:1187  [nsql3] 1700  MIGRATION job 722924980096368641: stepping through state running with error: <nil>
I211228 11:07:39.271152 24556 jobs/registry.go:1187  [nsql3] 1701  MIGRATION job 722924980096368641: stepping through state succeeded with error: <nil>
I211228 11:07:39.662146 22577 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 1702  set cluster version to: 21.2-34
I211228 11:07:39.662172 22859 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 1703  set cluster version to: 21.2-34
I211228 11:07:39.720047 22703 util/log/event_log.go:32  [nsql3,client=127.0.0.1:52566,hostssl,user=root] 1704 ={"Timestamp":1640689596656630910,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211228 11:07:39.824559 22793 2@rpc/context.go:742  [nsql3] 1705  closing [{n1: 127.0.0.1:34859 (default)} {n0: 127.0.0.1:34859 (default)}]
I211228 11:07:39.825551 22593 2@rpc/context.go:742  [nsql3] 1706  closing [{n0: 127.0.0.1:34859 (system)}]
I211228 11:07:39.826881 22414 2@rpc/context.go:742  [nsql2] 1707  closing [{n1: 127.0.0.1:34859 (default)} {n0: 127.0.0.1:34859 (default)}]
I211228 11:07:39.827674 22466 2@rpc/context.go:742  [nsql2] 1708  closing [{n0: 127.0.0.1:34859 (system)}]
W211228 11:07:39.828392 22855 jobs/registry.go:859  [nsql3] 1709  canceling all adopted jobs due to stopper quiescing
W211228 11:07:39.829010 22573 jobs/registry.go:859  [nsql2] 1710  canceling all adopted jobs due to stopper quiescing
W211228 11:07:39.829053 22681 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 1711  exiting heartbeat loop
W211228 11:07:39.830341 22766 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 1712  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211228 11:07:39.830508 22465 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 1713  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211228 11:07:39.832028 22431 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 1714  exiting heartbeat loop
    testing.go:1152: race detected during execution of test
I211228 11:07:39.839527 16663 testutils/testcluster/testcluster.go:106  [-] 1715  TestCluster quiescing nodes
W211228 11:07:39.843331 16923 jobs/registry.go:859  [n1] 1717  canceling all adopted jobs due to stopper quiescing
I211228 11:07:39.843194 17818 2@rpc/context.go:742  [n1] 1716  closing [{n1: 127.0.0.1:34859 (default)} {n0: 127.0.0.1:34859 (default)}]
W211228 11:07:39.843654 16883 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 1718  exiting heartbeat loop
I211228 11:07:39.940954 16689 kv/kvclient/kvcoord/transport_race.go:107  [-] 1719  transport race promotion: ran 221 iterations on up to 2997 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (72.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4010765&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4010765&tab=artifacts#/) on master @ [5ac733bb4927020bc1c52da24b2591742fde8e1f](https://github.com/cockroachdb/cockroach/commits/5ac733bb4927020bc1c52da24b2591742fde8e1f):


```
I211230 11:14:32.684681 22098 2@server/status/runtime_log.go:47  [nsql2] 1878  runtime stats: 902 MiB RSS, 481 goroutines (stacks: 5.2 MiB), 78 MiB/129 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 15 MiB, heap released: 4.0 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 67.6/4.6 %(u/s)time, 0.0 %gc (153x), 107 KiB/107 KiB (r/w)net
I211230 11:14:33.195703 22355 2@util/log/event_log.go:32  [nsql3] 1879 ={"Timestamp":1640862873195690049,"EventType":"runtime_stats","MemRSSBytes":945360896,"GoroutineCount":481,"MemStackSysBytes":4587520,"GoAllocBytes":50584328,"GoTotalBytes":135147544,"GoStatsStaleness":0.002140877,"HeapFragmentBytes":20817144,"HeapReservedBytes":45703168,"HeapReleasedBytes":4136960,"CGoAllocBytes":5972064,"CGoTotalBytes":13959168,"CGoCallRate":0.29998642,"CPUUserPercent":69.49685,"CPUSysPercent":4.4997964,"GCPausePercent":0.0057218107,"GCRunCount":154,"NetHostRecvBytes":108767,"NetHostSendBytes":108767}
I211230 11:14:33.196783 22355 2@server/status/runtime_log.go:47  [nsql3] 1880  runtime stats: 902 MiB RSS, 481 goroutines (stacks: 4.4 MiB), 48 MiB/129 MiB Go alloc/total(stale) (heap fragmentation: 20 MiB, heap reserved: 44 MiB, heap released: 3.9 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 69.5/4.5 %(u/s)time, 0.0 %gc (154x), 106 KiB/106 KiB (r/w)net
I211230 11:14:33.357942 16468 2@util/log/event_log.go:32  [n1] 1881 ={"Timestamp":1640862873357930559,"EventType":"runtime_stats","MemRSSBytes":945463296,"GoroutineCount":481,"MemStackSysBytes":4882432,"GoAllocBytes":53539040,"GoTotalBytes":135147544,"GoStatsStaleness":0.001730511,"HeapFragmentBytes":20115232,"HeapReservedBytes":43155456,"HeapReleasedBytes":4136960,"CGoAllocBytes":5972064,"CGoTotalBytes":13959168,"CGoCallRate":0.29999465,"CPUUserPercent":69.19876,"CPUSysPercent":4.49992,"GCPausePercent":0.0057219677,"GCRunCount":154,"NetHostRecvBytes":100559,"NetHostSendBytes":100559}
I211230 11:14:33.359109 16468 2@server/status/runtime_log.go:47  [n1] 1882  runtime stats: 902 MiB RSS, 481 goroutines (stacks: 4.7 MiB), 51 MiB/129 MiB Go alloc/total(stale) (heap fragmentation: 19 MiB, heap reserved: 41 MiB, heap released: 3.9 MiB), 5.7 MiB/13 MiB CGO alloc/total (0.3 CGO/sec), 69.2/4.5 %(u/s)time, 0.0 %gc (154x), 98 KiB/98 KiB (r/w)net
I211230 11:14:34.674034 16398 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 1883  [JOB 6] WAL created 000007
I211230 11:14:42.684318 22098 2@util/log/event_log.go:32  [nsql2] 1884 ={"Timestamp":1640862882684302815,"EventType":"runtime_stats","MemRSSBytes":959074304,"GoroutineCount":481,"MemStackSysBytes":4849664,"GoAllocBytes":52359528,"GoTotalBytes":139046936,"GoStatsStaleness":0.001835339,"HeapFragmentBytes":20418200,"HeapReservedBytes":47964160,"HeapReleasedBytes":4431872,"CGoAllocBytes":10166496,"CGoTotalBytes":18153472,"CGoCallRate":0.39996305,"CPUUserPercent":64.993996,"CPUSysPercent":5.5994825,"GCPausePercent":0.005695354,"GCRunCount":158,"NetHostRecvBytes":97838,"NetHostSendBytes":97838}
I211230 11:14:42.685344 22098 2@server/status/runtime_log.go:47  [nsql2] 1885  runtime stats: 915 MiB RSS, 481 goroutines (stacks: 4.6 MiB), 50 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 20 MiB, heap reserved: 46 MiB, heap released: 4.2 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 65.0/5.6 %(u/s)time, 0.0 %gc (158x), 96 KiB/96 KiB (r/w)net
I211230 11:14:43.195969 22355 2@util/log/event_log.go:32  [nsql3] 1886 ={"Timestamp":1640862883195953646,"EventType":"runtime_stats","MemRSSBytes":957005824,"GoroutineCount":481,"MemStackSysBytes":5046272,"GoAllocBytes":64190816,"GoTotalBytes":139046936,"GoStatsStaleness":0.001727522,"HeapFragmentBytes":18998944,"HeapReservedBytes":37355520,"HeapReleasedBytes":4431872,"CGoAllocBytes":10166496,"CGoTotalBytes":18153472,"CGoCallRate":0.3999892,"CPUUserPercent":63.89827,"CPUSysPercent":5.499851,"GCPausePercent":0.004768691,"GCRunCount":158,"NetHostRecvBytes":91909,"NetHostSendBytes":91909}
I211230 11:14:43.197077 22355 2@server/status/runtime_log.go:47  [nsql3] 1887  runtime stats: 913 MiB RSS, 481 goroutines (stacks: 4.8 MiB), 61 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 36 MiB, heap released: 4.2 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 63.9/5.5 %(u/s)time, 0.0 %gc (158x), 90 KiB/90 KiB (r/w)net
I211230 11:14:43.353217 16447 2@gossip/gossip.go:566  [n1] 1888  gossip status (ok, 1 node)
I211230 11:14:43.353217 16447 2@gossip/gossip.go:566  [n1] 1888 +gossip client (0/3 cur/max conns)
I211230 11:14:43.353217 16447 2@gossip/gossip.go:566  [n1] 1888 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211230 11:14:43.353217 16447 2@gossip/gossip.go:566  [n1] 1888 +gossip connectivity
I211230 11:14:43.353217 16447 2@gossip/gossip.go:566  [n1] 1888 +  n1 [sentinel];
I211230 11:14:43.359332 16468 2@util/log/event_log.go:32  [n1] 1889 ={"Timestamp":1640862883359315543,"EventType":"runtime_stats","MemRSSBytes":956399616,"GoroutineCount":481,"MemStackSysBytes":5210112,"GoAllocBytes":65034392,"GoTotalBytes":139046936,"GoStatsStaleness":0.002537764,"HeapFragmentBytes":18818920,"HeapReservedBytes":36528128,"HeapReleasedBytes":4431872,"CGoAllocBytes":10166496,"CGoTotalBytes":18153472,"CGoCallRate":0.3999449,"CPUUserPercent":63.291283,"CPUSysPercent":5.4992423,"GCPausePercent":0.004768163,"GCRunCount":158,"NetHostRecvBytes":92625,"NetHostSendBytes":92625}
I211230 11:14:43.360871 16468 2@server/status/runtime_log.go:47  [n1] 1890  runtime stats: 912 MiB RSS, 481 goroutines (stacks: 5.0 MiB), 62 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 35 MiB, heap released: 4.2 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.4 CGO/sec), 63.3/5.5 %(u/s)time, 0.0 %gc (158x), 90 KiB/90 KiB (r/w)net
I211230 11:14:47.644545 17398 server/auto_upgrade.go:44  [n1] 1891  auto upgrade disabled by testing
I211230 11:14:52.685128 22098 2@util/log/event_log.go:32  [nsql2] 1892 ={"Timestamp":1640862892685113865,"EventType":"runtime_stats","MemRSSBytes":977072128,"GoroutineCount":480,"MemStackSysBytes":5111808,"GoAllocBytes":82727120,"GoTotalBytes":140972056,"GoStatsStaleness":0.001813185,"HeapFragmentBytes":17968944,"HeapReservedBytes":21708800,"HeapReleasedBytes":2506752,"CGoAllocBytes":10166496,"CGoTotalBytes":18153472,"CGoCallRate":0.29997566,"CPUUserPercent":73.39405,"CPUSysPercent":5.1995783,"GCPausePercent":0.007696906,"GCRunCount":162,"NetHostRecvBytes":91026,"NetHostSendBytes":91026}
I211230 11:14:52.686229 22098 2@server/status/runtime_log.go:47  [nsql2] 1893  runtime stats: 932 MiB RSS, 480 goroutines (stacks: 4.9 MiB), 79 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 21 MiB, heap released: 2.4 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.3 CGO/sec), 73.4/5.2 %(u/s)time, 0.0 %gc (162x), 89 KiB/89 KiB (r/w)net
I211230 11:14:52.729357 22400 jobs/adopt.go:106  [nsql3] 1894  claimed 1 jobs
I211230 11:14:52.747580 24051 jobs/adopt.go:247  [nsql3] 1895  job 723492642216738817: resuming execution
I211230 11:14:52.807944 24055 jobs/registry.go:1187  [nsql3] 1896  MIGRATION job 723492642216738817: stepping through state running with error: <nil>
I211230 11:14:52.848827 24055 jobs/registry.go:1187  [nsql3] 1897  MIGRATION job 723492642216738817: stepping through state succeeded with error: <nil>
I211230 11:14:52.893801 24059 jobs/registry.go:1187  [nsql2] 1898  AUTO CREATE STATS job 723492840349859842: stepping through state running with error: <nil>
I211230 11:14:53.045938 24059 jobs/registry.go:1187  [nsql2] 1899  AUTO CREATE STATS job 723492840349859842: stepping through state succeeded with error: <nil>
I211230 11:14:53.162984 22402 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 1900  set cluster version to: 21.2-34
I211230 11:14:53.163238 22154 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 1901  set cluster version to: 21.2-34
I211230 11:14:53.196977 22355 2@util/log/event_log.go:32  [nsql3] 1902 ={"Timestamp":1640862893196961793,"EventType":"runtime_stats","MemRSSBytes":913539072,"GoroutineCount":482,"MemStackSysBytes":5898240,"GoAllocBytes":91511880,"GoTotalBytes":140980248,"GoStatsStaleness":0.002466372,"HeapFragmentBytes":14558136,"HeapReservedBytes":15556608,"HeapReleasedBytes":2498560,"CGoAllocBytes":10166496,"CGoTotalBytes":18153472,"CGoCallRate":0.29996982,"CPUUserPercent":83.991554,"CPUSysPercent":6.399356,"GCPausePercent":0.008962458,"GCRunCount":163,"NetHostRecvBytes":209925,"NetHostSendBytes":209925}
I211230 11:14:53.198190 22355 2@server/status/runtime_log.go:47  [nsql3] 1903  runtime stats: 871 MiB RSS, 482 goroutines (stacks: 5.6 MiB), 87 MiB/134 MiB Go alloc/total(stale) (heap fragmentation: 14 MiB, heap reserved: 15 MiB, heap released: 2.4 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.3 CGO/sec), 84.0/6.4 %(u/s)time, 0.0 %gc (163x), 205 KiB/205 KiB (r/w)net
I211230 11:14:53.219200 22470 util/log/event_log.go:32  [nsql3,client=127.0.0.1:40778,hostssl,user=root] 1904 ={"Timestamp":1640862833322564910,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211230 11:14:53.351898 22373 2@rpc/context.go:742  [nsql3] 1905  closing [{n1: 127.0.0.1:41457 (default)} {n0: 127.0.0.1:41457 (default)}]
I211230 11:14:53.352170 22338 2@rpc/context.go:742  [nsql3] 1906  closing [{n0: 127.0.0.1:41457 (system)}]
W211230 11:14:53.352684 22398 jobs/registry.go:859  [nsql3] 1907  canceling all adopted jobs due to stopper quiescing
I211230 11:14:53.352711 22116 2@rpc/context.go:742  [nsql2] 1908  closing [{n1: 127.0.0.1:41457 (default)} {n0: 127.0.0.1:41457 (default)}]
I211230 11:14:53.353502 22066 2@rpc/context.go:742  [nsql2] 1909  closing [{n0: 127.0.0.1:41457 (system)}]
W211230 11:14:53.354258 22150 jobs/registry.go:859  [nsql2] 1910  canceling all adopted jobs due to stopper quiescing
W211230 11:14:53.354901 22341 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 1911  exiting heartbeat loop
W211230 11:14:53.356526 22099 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 1912  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211230 11:14:53.356628 22356 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 1913  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:34622->127.0.0.1:41457: use of closed network connection
W211230 11:14:53.358596 22009 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 1914  exiting heartbeat loop
I211230 11:14:53.363540 16468 2@util/log/event_log.go:32  [n1] 1915 ={"Timestamp":1640862893363530998,"EventType":"runtime_stats","MemRSSBytes":888442880,"GoroutineCount":258,"MemStackSysBytes":3407872,"GoAllocBytes":68821288,"GoTotalBytes":141463576,"GoStatsStaleness":0.001599444,"HeapFragmentBytes":18292440,"HeapReservedBytes":37486592,"HeapReleasedBytes":2015232,"CGoAllocBytes":10166496,"CGoTotalBytes":18153472,"CGoCallRate":0.29987335,"CPUUserPercent":89.36226,"CPUSysPercent":6.8970876,"GCPausePercent":0.010958452,"GCRunCount":164,"NetHostRecvBytes":249897,"NetHostSendBytes":249897}
I211230 11:14:53.364841 16468 2@server/status/runtime_log.go:47  [n1] 1916  runtime stats: 847 MiB RSS, 258 goroutines (stacks: 3.3 MiB), 66 MiB/135 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 36 MiB, heap released: 1.9 MiB), 9.7 MiB/17 MiB CGO alloc/total (0.3 CGO/sec), 89.4/6.9 %(u/s)time, 0.0 %gc (164x), 244 KiB/244 KiB (r/w)net
    testing.go:1152: race detected during execution of test
I211230 11:14:53.365806 16261 testutils/testcluster/testcluster.go:106  [-] 1917  TestCluster quiescing nodes
I211230 11:14:53.366669 16208 kv/kvclient/kvcoord/transport_race.go:107  [-] 1918  transport race promotion: ran 215 iterations on up to 3015 requests
I211230 11:14:53.366690 17389 2@rpc/context.go:742  [n1] 1919  closing [{n1: 127.0.0.1:41457 (default)} {n0: 127.0.0.1:41457 (default)}]
W211230 11:14:53.368192 16497 jobs/registry.go:859  [n1] 1920  canceling all adopted jobs due to stopper quiescing
W211230 11:14:53.371518 16503 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 1921  exiting heartbeat loop
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (69.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4017383&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4017383&tab=artifacts#/) on master @ [3471dc50db8406f3faa638940492dbe0cab53435](https://github.com/cockroachdb/cockroach/commits/3471dc50db8406f3faa638940492dbe0cab53435):


```
I211231 10:59:59.085355 18567 2@server/status/runtime_log.go:47  [nsql3] 885  runtime stats: 1.1 GiB RSS, 477 goroutines (stacks: 4.5 MiB), 75 MiB/186 MiB Go alloc/total(stale) (heap fragmentation: 25 MiB, heap reserved: 66 MiB, heap released: 2.2 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 66.8/4.7 %(u/s)time, 0.0 %gc (124x), 99 KiB/99 KiB (r/w)net
I211231 11:00:03.570386 14641 2@util/log/event_log.go:32  [n1] 886 ={"Timestamp":1640948403570374595,"EventType":"runtime_stats","MemRSSBytes":1201217536,"GoroutineCount":478,"MemStackSysBytes":5177344,"GoAllocBytes":97350400,"GoTotalBytes":194829352,"GoStatsStaleness":0.001992429,"HeapFragmentBytes":23350528,"HeapReservedBytes":52576256,"HeapReleasedBytes":1900544,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.29998463,"CPUUserPercent":72.496284,"CPUSysPercent":4.899749,"GCPausePercent":0.003942548,"GCRunCount":125,"NetHostRecvBytes":100900,"NetHostSendBytes":100900}
I211231 11:00:03.571486 14641 2@server/status/runtime_log.go:47  [n1] 887  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 4.9 MiB), 93 MiB/186 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 50 MiB, heap released: 1.8 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 72.5/4.9 %(u/s)time, 0.0 %gc (125x), 98 KiB/98 KiB (r/w)net
I211231 11:00:06.707696 18505 sql/sqlliveness/slstorage/slstorage.go:388  [nsql3] 888  deleted 1 expired SQL liveness sessions
I211231 11:00:08.534841 18261 2@util/log/event_log.go:32  [nsql2] 889 ={"Timestamp":1640948408534828069,"EventType":"runtime_stats","MemRSSBytes":1206145024,"GoroutineCount":478,"MemStackSysBytes":5898240,"GoAllocBytes":146089072,"GoTotalBytes":194829352,"GoStatsStaleness":0.001783575,"HeapFragmentBytes":21478288,"HeapReservedBytes":4988928,"HeapReleasedBytes":1900544,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.2999997,"CPUUserPercent":77.79993,"CPUSysPercent":5.399995,"GCPausePercent":0.0030661472,"GCRunCount":126,"NetHostRecvBytes":107862,"NetHostSendBytes":107862}
I211231 11:00:08.536001 18261 2@server/status/runtime_log.go:47  [nsql2] 890  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 5.6 MiB), 139 MiB/186 MiB Go alloc/total(stale) (heap fragmentation: 20 MiB, heap reserved: 4.8 MiB, heap released: 1.8 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 77.8/5.4 %(u/s)time, 0.0 %gc (126x), 105 KiB/105 KiB (r/w)net
I211231 11:00:09.085175 18567 2@util/log/event_log.go:32  [nsql3] 891 ={"Timestamp":1640948409085157406,"EventType":"runtime_stats","MemRSSBytes":1207865344,"GoroutineCount":478,"MemStackSysBytes":5013504,"GoAllocBytes":86701328,"GoTotalBytes":196738088,"GoStatsStaleness":0.001841927,"HeapFragmentBytes":24865520,"HeapReservedBytes":63782912,"HeapReleasedBytes":4186112,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.29997244,"CPUUserPercent":77.59287,"CPUSysPercent":5.399504,"GCPausePercent":0.003015683,"GCRunCount":127,"NetHostRecvBytes":105238,"NetHostSendBytes":105238}
I211231 11:00:09.086299 18567 2@server/status/runtime_log.go:47  [nsql3] 892  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 4.8 MiB), 83 MiB/188 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 61 MiB, heap released: 4.0 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 77.6/5.4 %(u/s)time, 0.0 %gc (127x), 103 KiB/103 KiB (r/w)net
I211231 11:00:13.571292 14641 2@util/log/event_log.go:32  [n1] 893 ={"Timestamp":1640948413571276702,"EventType":"runtime_stats","MemRSSBytes":1212256256,"GoroutineCount":478,"MemStackSysBytes":5341184,"GoAllocBytes":95276256,"GoTotalBytes":197245992,"GoStatsStaleness":0.002436203,"HeapFragmentBytes":23597856,"HeapReservedBytes":56655872,"HeapReleasedBytes":3678208,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.29997298,"CPUUserPercent":74.89326,"CPUSysPercent":4.99955,"GCPausePercent":0.003867952,"GCRunCount":128,"NetHostRecvBytes":102917,"NetHostSendBytes":102917}
I211231 11:00:13.572622 14641 2@server/status/runtime_log.go:47  [n1] 894  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 5.1 MiB), 91 MiB/188 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 54 MiB, heap released: 3.5 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 74.9/5.0 %(u/s)time, 0.0 %gc (128x), 100 KiB/100 KiB (r/w)net
I211231 11:00:18.535650 18261 2@util/log/event_log.go:32  [nsql2] 895 ={"Timestamp":1640948418535630867,"EventType":"runtime_stats","MemRSSBytes":1204346880,"GoroutineCount":478,"MemStackSysBytes":5767168,"GoAllocBytes":106717136,"GoTotalBytes":197778472,"GoStatsStaleness":0.001811646,"HeapFragmentBytes":23207984,"HeapReservedBytes":45711360,"HeapReleasedBytes":3145728,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.29997596,"CPUUserPercent":61.695057,"CPUSysPercent":3.5997114,"GCPausePercent":0.003917106,"GCRunCount":129,"NetHostRecvBytes":92943,"NetHostSendBytes":92943}
I211231 11:00:18.536702 18261 2@server/status/runtime_log.go:47  [nsql2] 896  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 5.5 MiB), 102 MiB/189 MiB Go alloc/total(stale) (heap fragmentation: 22 MiB, heap reserved: 44 MiB, heap released: 3.0 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 61.7/3.6 %(u/s)time, 0.0 %gc (129x), 91 KiB/91 KiB (r/w)net
I211231 11:00:19.086208 18567 2@util/log/event_log.go:32  [nsql3] 897 ={"Timestamp":1640948419086193363,"EventType":"runtime_stats","MemRSSBytes":1201545216,"GoroutineCount":478,"MemStackSysBytes":6029312,"GoAllocBytes":121155352,"GoTotalBytes":197778472,"GoStatsStaleness":0.001808914,"HeapFragmentBytes":21696744,"HeapReservedBytes":32522240,"HeapReleasedBytes":3145728,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.29996893,"CPUUserPercent":60.69371,"CPUSysPercent":3.3996477,"GCPausePercent":0.0029095486,"GCRunCount":129,"NetHostRecvBytes":94335,"NetHostSendBytes":94335}
I211231 11:00:19.087251 18567 2@server/status/runtime_log.go:47  [nsql3] 898  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 5.8 MiB), 116 MiB/189 MiB Go alloc/total(stale) (heap fragmentation: 21 MiB, heap reserved: 31 MiB, heap released: 3.0 MiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 60.7/3.4 %(u/s)time, 0.0 %gc (129x), 92 KiB/92 KiB (r/w)net
I211231 11:00:22.767961 18635 jobs/adopt.go:105  [nsql3] 899  claimed 1 jobs
I211231 11:00:22.790955 20200 jobs/adopt.go:240  [nsql3] 900  job 723772926118395905: resuming execution
I211231 11:00:22.850402 20119 jobs/registry.go:1186  [nsql3] 901  MIGRATION job 723772926118395905: stepping through state running with error: <nil>
I211231 11:00:22.912429 20119 jobs/registry.go:1186  [nsql3] 902  MIGRATION job 723772926118395905: stepping through state succeeded with error: <nil>
I211231 11:00:23.564532 14636 2@gossip/gossip.go:566  [n1] 903  gossip status (ok, 1 node)
I211231 11:00:23.564532 14636 2@gossip/gossip.go:566  [n1] 903 +gossip client (0/3 cur/max conns)
I211231 11:00:23.564532 14636 2@gossip/gossip.go:566  [n1] 903 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I211231 11:00:23.564532 14636 2@gossip/gossip.go:566  [n1] 903 +gossip connectivity
I211231 11:00:23.564532 14636 2@gossip/gossip.go:566  [n1] 903 +  n1 [sentinel];
I211231 11:00:23.571043 14641 2@util/log/event_log.go:32  [n1] 904 ={"Timestamp":1640948423571032819,"EventType":"runtime_stats","MemRSSBytes":1198403584,"GoroutineCount":478,"MemStackSysBytes":6193152,"GoAllocBytes":131445424,"GoTotalBytes":200309800,"GoStatsStaleness":0.001855179,"HeapFragmentBytes":18869584,"HeapReservedBytes":27426816,"HeapReleasedBytes":614400,"CGoAllocBytes":6048096,"CGoTotalBytes":12525568,"CGoCallRate":0.30000725,"CPUUserPercent":58.80142,"CPUSysPercent":4.0000963,"GCPausePercent":0.0027980576,"GCRunCount":130,"NetHostRecvBytes":128635,"NetHostSendBytes":128635}
I211231 11:00:23.572066 14641 2@server/status/runtime_log.go:47  [n1] 905  runtime stats: 1.1 GiB RSS, 478 goroutines (stacks: 5.9 MiB), 125 MiB/191 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 26 MiB, heap released: 600 KiB), 5.8 MiB/12 MiB CGO alloc/total (0.3 CGO/sec), 58.8/4.0 %(u/s)time, 0.0 %gc (130x), 126 KiB/126 KiB (r/w)net
I211231 11:00:23.655075 15591 server/auto_upgrade.go:44  [n1] 906  auto upgrade disabled by testing
I211231 11:00:23.658360 18389 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 907  set cluster version to: 21.2-34
I211231 11:00:23.658326 18637 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 908  set cluster version to: 21.2-34
I211231 11:00:23.725861 18676 util/log/event_log.go:32  [nsql3,client=127.0.0.1:46660,hostssl,user=root] 909 ={"Timestamp":1640948369256713139,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I211231 11:00:23.944416 18597 2@rpc/context.go:742  [nsql3] 910  closing [{n1: 127.0.0.1:43627 (default)} {n0: 127.0.0.1:43627 (default)}]
I211231 11:00:23.945870 18126 2@rpc/context.go:742  [nsql2] 911  closing [{n0: 127.0.0.1:43627 (default)}]
I211231 11:00:23.945986 18442 2@rpc/context.go:742  [nsql3] 912  closing [{n0: 127.0.0.1:43627 (system)}]
I211231 11:00:23.946375 18272 2@rpc/context.go:742  [nsql2] 913  closing [{n1: 127.0.0.1:43627 (default)} {n0: 127.0.0.1:43627 (default)}]
I211231 11:00:23.947316 18479 2@rpc/context.go:742  [nsql3] 914  closing [{n0: 127.0.0.1:43627 (system)}]
W211231 11:00:23.947498 18506 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 915  exiting heartbeat loop
W211231 11:00:23.947689 18262 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 916  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W211231 11:00:23.948149 18633 jobs/registry.go:858  [nsql3] 917  canceling all adopted jobs due to stopper quiescing
I211231 11:00:23.948725 18242 2@rpc/context.go:742  [nsql2] 918  closing [{n0: 127.0.0.1:43627 (system)}]
W211231 11:00:23.949348 18337 jobs/registry.go:858  [nsql2] 919  canceling all adopted jobs due to stopper quiescing
W211231 11:00:23.950149 18274 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 920  exiting heartbeat loop
W211231 11:00:23.951013 18568 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 921  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:46510->127.0.0.1:43627: use of closed network connection
    testing.go:1152: race detected during execution of test
I211231 11:00:23.959301 14443 testutils/testcluster/testcluster.go:106  [-] 922  TestCluster quiescing nodes
W211231 11:00:23.960335 20305 kv/kvserver/intentresolver/intent_resolver.go:793  [-] 923  failed to gc transaction record: could not GC completed transaction anchored at /Table/28/1/3/0: node unavailable; try another peer
I211231 11:00:23.961208 14486 2@rpc/context.go:742  [n1] 924  closing [{n1: 127.0.0.1:43627 (default)}]
W211231 11:00:23.961975 14558 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 925  exiting heartbeat loop
W211231 11:00:23.962979 14870 jobs/registry.go:858  [n1] 926  canceling all adopted jobs due to stopper quiescing
I211231 11:00:23.963801 15651 2@rpc/context.go:742  [n1] 927  closing [{n1: 127.0.0.1:43627 (default)} {n0: 127.0.0.1:43627 (default)}]
I211231 11:00:24.104011 14608 kv/kvclient/kvcoord/transport_race.go:107  [-] 928  transport race promotion: ran 183 iterations on up to 1685 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (59.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4022329&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4022329&tab=artifacts#/) on master @ [fa9480a8b661b31a3c5e4c3428595b3e7c320700](https://github.com/cockroachdb/cockroach/commits/fa9480a8b661b31a3c5e4c3428595b3e7c320700):


```
I220101 11:27:16.814819 15768 2@server/status/runtime_log.go:47  [nsql2] 777  runtime stats: 875 MiB RSS, 478 goroutines (stacks: 5.3 MiB), 77 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 19 MiB, heap released: 14 MiB), 5.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 68.2/4.7 %(u/s)time, 0.0 %gc (108x), 108 KiB/108 KiB (r/w)net
I220101 11:27:17.363064 16063 2@util/log/event_log.go:32  [nsql3] 778 ={"Timestamp":1641036437363047585,"EventType":"runtime_stats","MemRSSBytes":910708736,"GoroutineCount":478,"MemStackSysBytes":5996544,"GoAllocBytes":89937912,"GoTotalBytes":139213864,"GoStatsStaleness":0.001863883,"HeapFragmentBytes":15681544,"HeapReservedBytes":11714560,"HeapReleasedBytes":15081472,"CGoAllocBytes":5833696,"CGoTotalBytes":16449536,"CGoCallRate":0.29999244,"CPUUserPercent":67.898285,"CPUSysPercent":4.4998865,"GCPausePercent":0.005790044,"GCRunCount":108,"NetHostRecvBytes":108179,"NetHostSendBytes":108179}
I220101 11:27:17.364139 16063 2@server/status/runtime_log.go:47  [nsql3] 779  runtime stats: 868 MiB RSS, 478 goroutines (stacks: 5.7 MiB), 86 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 15 MiB, heap reserved: 11 MiB, heap released: 14 MiB), 5.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 67.9/4.5 %(u/s)time, 0.0 %gc (108x), 106 KiB/106 KiB (r/w)net
I220101 11:27:21.980068 11961 2@util/log/event_log.go:32  [n1] 780 ={"Timestamp":1641036441980055281,"EventType":"runtime_stats","MemRSSBytes":919736320,"GoroutineCount":478,"MemStackSysBytes":4718592,"GoAllocBytes":48917736,"GoTotalBytes":137673768,"GoStatsStaleness":0.00178216,"HeapFragmentBytes":20034328,"HeapReservedBytes":48119808,"HeapReleasedBytes":16621568,"CGoAllocBytes":5833696,"CGoTotalBytes":16449536,"CGoCallRate":0.29998246,"CPUUserPercent":71.6958,"CPUSysPercent":4.8997135,"GCPausePercent":0.008460565,"GCRunCount":111,"NetHostRecvBytes":102321,"NetHostSendBytes":102321}
I220101 11:27:21.981115 11961 2@server/status/runtime_log.go:47  [n1] 781  runtime stats: 877 MiB RSS, 478 goroutines (stacks: 4.5 MiB), 47 MiB/131 MiB Go alloc/total(stale) (heap fragmentation: 19 MiB, heap reserved: 46 MiB, heap released: 16 MiB), 5.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 71.7/4.9 %(u/s)time, 0.0 %gc (111x), 100 KiB/100 KiB (r/w)net
I220101 11:27:26.813910 15768 2@util/log/event_log.go:32  [nsql2] 782 ={"Timestamp":1641036446813896229,"EventType":"runtime_stats","MemRSSBytes":926392320,"GoroutineCount":478,"MemStackSysBytes":4882432,"GoAllocBytes":64706984,"GoTotalBytes":139115560,"GoStatsStaleness":0.001775124,"HeapFragmentBytes":18335320,"HeapReservedBytes":35307520,"HeapReleasedBytes":15179776,"CGoAllocBytes":5833696,"CGoTotalBytes":16449536,"CGoCallRate":0.29999065,"CPUUserPercent":69.297844,"CPUSysPercent":4.299866,"GCPausePercent":0.008976111,"GCRunCount":113,"NetHostRecvBytes":93185,"NetHostSendBytes":93185}
I220101 11:27:26.815009 15768 2@server/status/runtime_log.go:47  [nsql2] 783  runtime stats: 884 MiB RSS, 478 goroutines (stacks: 4.7 MiB), 62 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 18 MiB, heap reserved: 34 MiB, heap released: 14 MiB), 5.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 69.3/4.3 %(u/s)time, 0.0 %gc (113x), 91 KiB/91 KiB (r/w)net
I220101 11:27:27.364084 16063 2@util/log/event_log.go:32  [nsql3] 784 ={"Timestamp":1641036447364068816,"EventType":"runtime_stats","MemRSSBytes":927436800,"GoroutineCount":478,"MemStackSysBytes":5308416,"GoAllocBytes":73649736,"GoTotalBytes":139115560,"GoStatsStaleness":0.00176271,"HeapFragmentBytes":17134008,"HeapReservedBytes":27140096,"HeapReleasedBytes":15179776,"CGoAllocBytes":5833696,"CGoTotalBytes":16449536,"CGoCallRate":0.29996935,"CPUUserPercent":69.69288,"CPUSysPercent":4.2995605,"GCPausePercent":0.008975473,"GCRunCount":113,"NetHostRecvBytes":94233,"NetHostSendBytes":94233}
I220101 11:27:27.365175 16063 2@server/status/runtime_log.go:47  [nsql3] 785  runtime stats: 884 MiB RSS, 478 goroutines (stacks: 5.1 MiB), 70 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 16 MiB, heap reserved: 26 MiB, heap released: 14 MiB), 5.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 69.7/4.3 %(u/s)time, 0.0 %gc (113x), 92 KiB/92 KiB (r/w)net
I220101 11:27:28.795773 13094 server/auto_upgrade.go:44  [n1] 786  auto upgrade disabled by testing
I220101 11:27:31.975603 12157 2@gossip/gossip.go:566  [n1] 787  gossip status (ok, 1 node)
I220101 11:27:31.975603 12157 2@gossip/gossip.go:566  [n1] 787 +gossip client (0/3 cur/max conns)
I220101 11:27:31.975603 12157 2@gossip/gossip.go:566  [n1] 787 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I220101 11:27:31.975603 12157 2@gossip/gossip.go:566  [n1] 787 +gossip connectivity
I220101 11:27:31.975603 12157 2@gossip/gossip.go:566  [n1] 787 +  n1 [sentinel];
I220101 11:27:31.981562 11961 2@util/log/event_log.go:32  [n1] 788 ={"Timestamp":1641036451981549401,"EventType":"runtime_stats","MemRSSBytes":927535104,"GoroutineCount":478,"MemStackSysBytes":5275648,"GoAllocBytes":68410288,"GoTotalBytes":139525160,"GoStatsStaleness":0.002480204,"HeapFragmentBytes":17884240,"HeapReservedBytes":32071680,"HeapReleasedBytes":14770176,"CGoAllocBytes":5833696,"CGoTotalBytes":16449536,"CGoCallRate":0.29995528,"CPUUserPercent":65.6902,"CPUSysPercent":4.6992993,"GCPausePercent":0.0062169232,"GCRunCount":115,"NetHostRecvBytes":91886,"NetHostSendBytes":91886}
I220101 11:27:31.983086 11961 2@server/status/runtime_log.go:47  [n1] 789  runtime stats: 885 MiB RSS, 478 goroutines (stacks: 5.0 MiB), 65 MiB/133 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 31 MiB, heap released: 14 MiB), 5.6 MiB/16 MiB CGO alloc/total (0.3 CGO/sec), 65.7/4.7 %(u/s)time, 0.0 %gc (115x), 90 KiB/90 KiB (r/w)net
I220101 11:27:33.361624 12138 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 790  [JOB 6] WAL created 000007
I220101 11:27:36.696396 16164 jobs/adopt.go:105  [nsql3] 791  claimed 1 jobs
I220101 11:27:36.731595 17802 jobs/adopt.go:240  [nsql3] 792  job 724061377161887745: resuming execution
I220101 11:27:36.796432 17729 jobs/registry.go:1186  [nsql3] 793  MIGRATION job 724061377161887745: stepping through state running with error: <nil>
I220101 11:27:36.814263 15768 2@util/log/event_log.go:32  [nsql2] 794 ={"Timestamp":1641036456814250684,"EventType":"runtime_stats","MemRSSBytes":936120320,"GoroutineCount":479,"MemStackSysBytes":5079040,"GoAllocBytes":57762432,"GoTotalBytes":142539816,"GoStatsStaleness":0.001821147,"HeapFragmentBytes":17800576,"HeapReservedBytes":46014464,"HeapReleasedBytes":11755520,"CGoAllocBytes":10028128,"CGoTotalBytes":20643840,"CGoCallRate":0.3999857,"CPUUserPercent":72.99739,"CPUSysPercent":5.8997893,"GCPausePercent":0.006315914,"GCRunCount":118,"NetHostRecvBytes":109777,"NetHostSendBytes":109777}
I220101 11:27:36.815447 15768 2@server/status/runtime_log.go:47  [nsql2] 795  runtime stats: 893 MiB RSS, 479 goroutines (stacks: 4.8 MiB), 55 MiB/136 MiB Go alloc/total(stale) (heap fragmentation: 17 MiB, heap reserved: 44 MiB, heap released: 11 MiB), 9.6 MiB/20 MiB CGO alloc/total (0.4 CGO/sec), 73.0/5.9 %(u/s)time, 0.0 %gc (118x), 107 KiB/107 KiB (r/w)net
I220101 11:27:36.855631 17729 jobs/registry.go:1186  [nsql3] 796  MIGRATION job 724061377161887745: stepping through state succeeded with error: <nil>
I220101 11:27:36.955925 16166 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 797  set cluster version to: 21.2-34
I220101 11:27:36.956730 15853 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 798  set cluster version to: 21.2-34
I220101 11:27:37.014002 16170 util/log/event_log.go:32  [nsql3,client=127.0.0.1:36074,hostssl,user=root] 799 ={"Timestamp":1641036397553449829,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I220101 11:27:37.071481 17911 jobs/registry.go:1186  [nsql2] 800  AUTO CREATE STATS job 724061575443873794: stepping through state running with error: <nil>
I220101 11:27:37.193741 15690 2@rpc/context.go:742  [nsql2] 801  closing [{n1: 127.0.0.1:33149 (default)} {n0: 127.0.0.1:33149 (default)}]
W220101 11:27:37.194617 15849 jobs/registry.go:858  [nsql2] 803  canceling all adopted jobs due to stopper quiescing
I220101 11:27:37.194439 15912 2@rpc/context.go:742  [nsql3] 802  closing [{n0: 127.0.0.1:33149 (system)}]
I220101 11:27:37.194954 15981 2@rpc/context.go:742  [nsql3] 804  closing [{n1: 127.0.0.1:33149 (default)} {n0: 127.0.0.1:33149 (default)}]
I220101 11:27:37.195420 15751 2@rpc/context.go:742  [nsql2] 805  closing [{n0: 127.0.0.1:33149 (system)}]
I220101 11:27:37.195951 15993 2@rpc/context.go:742  [nsql3] 806  closing [{n0: 127.0.0.1:33149 (system)}]
W220101 11:27:37.196866 16162 jobs/registry.go:858  [nsql3] 807  canceling all adopted jobs due to stopper quiescing
W220101 11:27:37.197538 16082 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 808  exiting heartbeat loop
W220101 11:27:37.197499 17911 kv/txn.go:712  [nsql2,job=724061575443873794] 809  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W220101 11:27:37.199353 16064 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 810  error consuming GossipSubscription RPC: rpc error: code = Unavailable desc = error reading from server: read tcp 127.0.0.1:57660->127.0.0.1:33149: use of closed network connection
W220101 11:27:37.199447 17911 kv/txn.go:712  [nsql2,job=724061575443873794] 811  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
W220101 11:27:37.201273 15674 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 812  exiting heartbeat loop
W220101 11:27:37.201602 17911 kv/txn.go:712  [nsql2,job=724061575443873794] 813  failure aborting transaction: node unavailable; try another peer; abort caused by: job 724061575443873794: context canceled
W220101 11:27:37.202322 15769 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 814  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W220101 11:27:37.206608 17107 sql/stats/automatic_stats.go:488  [nsql2] 815  failed to create statistics on table 52: create-stats: job 724061575443873794: node liveness error: restarting in background
    testing.go:1152: race detected during execution of test
I220101 11:27:37.212572 11899 testutils/testcluster/testcluster.go:106  [-] 816  TestCluster quiescing nodes
I220101 11:27:37.213413 13104 2@rpc/context.go:742  [n1] 817  closing [{n1: 127.0.0.1:33149 (default)} {n0: 127.0.0.1:33149 (default)}]
W220101 11:27:37.214227 12360 jobs/registry.go:858  [n1] 818  canceling all adopted jobs due to stopper quiescing
W220101 11:27:37.216522 12197 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 819  exiting heartbeat loop
I220101 11:27:37.487793 11950 kv/kvclient/kvcoord/transport_race.go:107  [-] 820  transport race promotion: ran 207 iterations on up to 1766 requests
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (64.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/kvccl/kvtenantccl.TestTenantUpgradeFailure [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4033172&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4033172&tab=artifacts#/) on master @ [e3be114fc6fe8eda07aafbc7ee43cc452456f270](https://github.com/cockroachdb/cockroach/commits/e3be114fc6fe8eda07aafbc7ee43cc452456f270):


```
I220104 10:53:51.652885 16207 2@server/status/runtime_log.go:47  [nsql2] 804  runtime stats: 1.0 GiB RSS, 476 goroutines (stacks: 5.7 MiB), 113 MiB/163 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 6.2 MiB, heap released: 16 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 69.5/4.3 %(u/s)time, 0.0 %gc (91x), 89 KiB/89 KiB (r/w)net
I220104 10:53:52.209509 16536 2@util/log/event_log.go:32  [nsql3] 805 ={"Timestamp":1641293632209493953,"EventType":"runtime_stats","MemRSSBytes":1088413696,"GoroutineCount":477,"MemStackSysBytes":5373952,"GoAllocBytes":71998552,"GoTotalBytes":173005864,"GoStatsStaleness":0.001824524,"HeapFragmentBytes":25543592,"HeapReservedBytes":53977088,"HeapReleasedBytes":15073280,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.2999708,"CPUUserPercent":71.09308,"CPUSysPercent":4.3995714,"GCPausePercent":0.008132098,"GCRunCount":92,"NetHostRecvBytes":92816,"NetHostSendBytes":92816}
I220104 10:53:52.210602 16536 2@server/status/runtime_log.go:47  [nsql3] 806  runtime stats: 1.0 GiB RSS, 477 goroutines (stacks: 5.1 MiB), 69 MiB/165 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 52 MiB, heap released: 14 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 71.1/4.4 %(u/s)time, 0.0 %gc (92x), 91 KiB/91 KiB (r/w)net
I220104 10:53:57.063170 12628 2@util/log/event_log.go:32  [n1] 807 ={"Timestamp":1641293637063158701,"EventType":"runtime_stats","MemRSSBytes":1078620160,"GoroutineCount":478,"MemStackSysBytes":5734400,"GoAllocBytes":102972168,"GoTotalBytes":173022248,"GoStatsStaleness":0.001784831,"HeapFragmentBytes":24061176,"HeapReservedBytes":24141824,"HeapReleasedBytes":15056896,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.29996452,"CPUUserPercent":71.291565,"CPUSysPercent":4.199503,"GCPausePercent":0.0054115397,"GCRunCount":93,"NetHostRecvBytes":94599,"NetHostSendBytes":94599}
I220104 10:53:57.064236 12628 2@server/status/runtime_log.go:47  [n1] 808  runtime stats: 1.0 GiB RSS, 478 goroutines (stacks: 5.5 MiB), 98 MiB/165 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 23 MiB, heap released: 14 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 71.3/4.2 %(u/s)time, 0.0 %gc (93x), 92 KiB/92 KiB (r/w)net
I220104 10:54:01.652424 16207 2@util/log/event_log.go:32  [nsql2] 809 ={"Timestamp":1641293641652411179,"EventType":"runtime_stats","MemRSSBytes":1099948032,"GoroutineCount":478,"MemStackSysBytes":5210112,"GoAllocBytes":90430720,"GoTotalBytes":173898792,"GoStatsStaleness":0.001748073,"HeapFragmentBytes":25166592,"HeapReservedBytes":36978688,"HeapReleasedBytes":14180352,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.29998162,"CPUUserPercent":72.39557,"CPUSysPercent":5.1996818,"GCPausePercent":0.006942685,"GCRunCount":95,"NetHostRecvBytes":94599,"NetHostSendBytes":94599}
I220104 10:54:01.653455 16207 2@server/status/runtime_log.go:47  [nsql2] 810  runtime stats: 1.0 GiB RSS, 478 goroutines (stacks: 5.0 MiB), 86 MiB/166 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 35 MiB, heap released: 14 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 72.4/5.2 %(u/s)time, 0.0 %gc (95x), 92 KiB/92 KiB (r/w)net
I220104 10:54:02.209746 16536 2@util/log/event_log.go:32  [nsql3] 811 ={"Timestamp":1641293642209733785,"EventType":"runtime_stats","MemRSSBytes":1093709824,"GoroutineCount":479,"MemStackSysBytes":5734400,"GoAllocBytes":101106240,"GoTotalBytes":173898792,"GoStatsStaleness":0.001790106,"HeapFragmentBytes":24034752,"HeapReservedBytes":26910720,"HeapReleasedBytes":14180352,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.2999927,"CPUUserPercent":70.79828,"CPUSysPercent":5.099876,"GCPausePercent":0.0033047893,"GCRunCount":95,"NetHostRecvBytes":91639,"NetHostSendBytes":91639}
I220104 10:54:02.210939 16536 2@server/status/runtime_log.go:47  [nsql3] 812  runtime stats: 1.0 GiB RSS, 479 goroutines (stacks: 5.5 MiB), 96 MiB/166 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 26 MiB, heap released: 14 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 70.8/5.1 %(u/s)time, 0.0 %gc (95x), 90 KiB/90 KiB (r/w)net
I220104 10:54:02.400959 15967 sql/sqlliveness/slstorage/slstorage.go:388  [nsql2] 813  deleted 1 expired SQL liveness sessions
I220104 10:54:07.064205 12628 2@util/log/event_log.go:32  [n1] 814 ={"Timestamp":1641293647064193175,"EventType":"runtime_stats","MemRSSBytes":1100271616,"GoroutineCount":478,"MemStackSysBytes":5111808,"GoAllocBytes":72189528,"GoTotalBytes":176528424,"GoStatsStaleness":0.001872099,"HeapFragmentBytes":25901480,"HeapReservedBytes":57212928,"HeapReleasedBytes":11550720,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.299969,"CPUUserPercent":71.99256,"CPUSysPercent":5.1994624,"GCPausePercent":0.0043868865,"GCRunCount":97,"NetHostRecvBytes":94715,"NetHostSendBytes":94715}
I220104 10:54:07.065277 12628 2@server/status/runtime_log.go:47  [n1] 815  runtime stats: 1.0 GiB RSS, 478 goroutines (stacks: 4.9 MiB), 69 MiB/168 MiB Go alloc/total(stale) (heap fragmentation: 25 MiB, heap reserved: 55 MiB, heap released: 11 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 72.0/5.2 %(u/s)time, 0.0 %gc (97x), 92 KiB/92 KiB (r/w)net
I220104 10:54:11.653579 16207 2@util/log/event_log.go:32  [nsql2] 816 ={"Timestamp":1641293651653564826,"EventType":"runtime_stats","MemRSSBytes":1106907136,"GoroutineCount":478,"MemStackSysBytes":5570560,"GoAllocBytes":90884608,"GoTotalBytes":176536616,"GoStatsStaleness":0.001771696,"HeapFragmentBytes":24860160,"HeapReservedBytes":39108608,"HeapReleasedBytes":11542528,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.29996538,"CPUUserPercent":61.79287,"CPUSysPercent":3.89955,"GCPausePercent":0.0037262598,"GCRunCount":98,"NetHostRecvBytes":92119,"NetHostSendBytes":92119}
I220104 10:54:11.654700 16207 2@server/status/runtime_log.go:47  [nsql2] 817  runtime stats: 1.0 GiB RSS, 478 goroutines (stacks: 5.3 MiB), 87 MiB/168 MiB Go alloc/total(stale) (heap fragmentation: 24 MiB, heap reserved: 37 MiB, heap released: 11 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 61.8/3.9 %(u/s)time, 0.0 %gc (98x), 90 KiB/90 KiB (r/w)net
I220104 10:54:12.210579 16536 2@util/log/event_log.go:32  [nsql3] 818 ={"Timestamp":1641293652210564250,"EventType":"runtime_stats","MemRSSBytes":1098858496,"GoroutineCount":478,"MemStackSysBytes":5799936,"GoAllocBytes":103376312,"GoTotalBytes":176536616,"GoStatsStaleness":0.001873947,"HeapFragmentBytes":23657032,"HeapReservedBytes":27590656,"HeapReleasedBytes":11542528,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.29997516,"CPUUserPercent":62.7948,"CPUSysPercent":3.899677,"GCPausePercent":0.0037263813,"GCRunCount":98,"NetHostRecvBytes":91757,"NetHostSendBytes":91757}
I220104 10:54:12.211586 16536 2@server/status/runtime_log.go:47  [nsql3] 819  runtime stats: 1.0 GiB RSS, 478 goroutines (stacks: 5.5 MiB), 99 MiB/168 MiB Go alloc/total(stale) (heap fragmentation: 23 MiB, heap reserved: 26 MiB, heap released: 11 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 62.8/3.9 %(u/s)time, 0.0 %gc (98x), 90 KiB/90 KiB (r/w)net
I220104 10:54:17.057059 12607 2@gossip/gossip.go:566  [n1] 820  gossip status (ok, 1 node)
I220104 10:54:17.057059 12607 2@gossip/gossip.go:566  [n1] 820 +gossip client (0/3 cur/max conns)
I220104 10:54:17.057059 12607 2@gossip/gossip.go:566  [n1] 820 +gossip server (0/3 cur/max conns, infos 0/0 sent/received, bytes 0B/0B sent/received)
I220104 10:54:17.057059 12607 2@gossip/gossip.go:566  [n1] 820 +gossip connectivity
I220104 10:54:17.057059 12607 2@gossip/gossip.go:566  [n1] 820 +  n1 [sentinel];
I220104 10:54:17.066103 12628 2@util/log/event_log.go:32  [n1] 821 ={"Timestamp":1641293657066088892,"EventType":"runtime_stats","MemRSSBytes":1113980928,"GoroutineCount":478,"MemStackSysBytes":4816896,"GoAllocBytes":67344144,"GoTotalBytes":178478120,"GoStatsStaleness":0.002733247,"HeapFragmentBytes":26904816,"HeapReservedBytes":63299584,"HeapReleasedBytes":9601024,"CGoAllocBytes":5990752,"CGoTotalBytes":15474688,"CGoCallRate":0.2999432,"CPUUserPercent":60.188602,"CPUSysPercent":4.199205,"GCPausePercent":0.0045341514,"GCRunCount":100,"NetHostRecvBytes":81382,"NetHostSendBytes":81382}
I220104 10:54:17.067787 12628 2@server/status/runtime_log.go:47  [n1] 822  runtime stats: 1.0 GiB RSS, 478 goroutines (stacks: 4.6 MiB), 64 MiB/170 MiB Go alloc/total(stale) (heap fragmentation: 26 MiB, heap reserved: 60 MiB, heap released: 9.2 MiB), 5.7 MiB/15 MiB CGO alloc/total (0.3 CGO/sec), 60.2/4.2 %(u/s)time, 0.0 %gc (100x), 80 KiB/80 KiB (r/w)net
I220104 10:54:18.096389 16340 jobs/adopt.go:105  [nsql2] 823  claimed 1 jobs
I220104 10:54:18.119382 18226 jobs/adopt.go:240  [nsql2] 824  job 724904186053623809: resuming execution
I220104 10:54:18.179607 18172 jobs/registry.go:1093  [nsql2] 825  MIGRATION job 724904186053623809: stepping through state running with error: <nil>
I220104 10:54:18.285855 18172 jobs/registry.go:1093  [nsql2] 826  MIGRATION job 724904186053623809: stepping through state succeeded with error: <nil>
I220104 10:54:18.509992 12550 3@vendor/github.com/cockroachdb/pebble/event.go:664  [n1,pebble,s1] 827  [JOB 6] WAL created 000007
I220104 10:54:18.634038 16693 jobs/wait.go:117  [nsql3,client=127.0.0.1:60556,hostssl,user=root,migration-mgr] 828  waited for 1 [724904186053623809] queued jobs to complete 56.101198009s
I220104 10:54:18.663946 16342 server/settingswatcher/settings_watcher.go:209  [nsql2,rangefeed=settings] 829  set cluster version to: 21.2-34
I220104 10:54:18.664143 16560 server/settingswatcher/settings_watcher.go:209  [nsql3,rangefeed=settings] 830  set cluster version to: 21.2-34
I220104 10:54:18.713839 16693 util/log/event_log.go:32  [nsql3,client=127.0.0.1:60556,hostssl,user=root] 831 ={"Timestamp":1641293602390829530,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","PlaceholderValues":["'21.2-34'"],"SettingName":"version","Value":"21.2-34"}
I220104 10:54:18.842411 16261 2@rpc/context.go:742  [nsql2] 832  closing [{n1: 127.0.0.1:41257 (default)} {n0: 127.0.0.1:41257 (default)}]
W220104 10:54:18.844186 16338 jobs/registry.go:765  [nsql2] 833  canceling all adopted jobs due to stopper quiescing
I220104 10:54:18.844550 16210 2@rpc/context.go:742  [nsql2] 835  closing [{n0: 127.0.0.1:41257 (system)}]
W220104 10:54:18.844195 16556 jobs/registry.go:765  [nsql3] 834  canceling all adopted jobs due to stopper quiescing
I220104 10:54:18.845210 16365 2@rpc/context.go:742  [nsql3] 837  closing [{n1: 127.0.0.1:41257 (default)}]
I220104 10:54:18.844195 16492 2@rpc/context.go:742  [nsql3] 836  closing [{n0: 127.0.0.1:41257 (system)}]
I220104 10:54:18.845480 16596 2@rpc/context.go:742  [nsql3] 838  closing [{n1: 127.0.0.1:41257 (default)} {n0: 127.0.0.1:41257 (default)}]
W220104 10:54:18.846723 16208 ccl/kvccl/kvtenantccl/connector.go:183  [nsql2,tenant-connector] 839  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W220104 10:54:18.848683 16537 ccl/kvccl/kvtenantccl/connector.go:183  [nsql3,tenant-connector] 840  error consuming GossipSubscription RPC: rpc error: code = Canceled desc = context canceled
W220104 10:54:18.849001 15968 sql/sqlliveness/slinstance/slinstance.go:224  [nsql2] 841  exiting heartbeat loop
W220104 10:54:18.849567 16497 sql/sqlliveness/slinstance/slinstance.go:224  [nsql3] 842  exiting heartbeat loop
    testing.go:1152: race detected during execution of test
I220104 10:54:18.856854 12455 testutils/testcluster/testcluster.go:106  [-] 843  TestCluster quiescing nodes
I220104 10:54:18.858013 12523 kv/kvclient/kvcoord/transport_race.go:107  [-] 844  transport race promotion: ran 180 iterations on up to 1696 requests
W220104 10:54:18.859375 12835 jobs/registry.go:765  [n1] 845  canceling all adopted jobs due to stopper quiescing
I220104 10:54:18.859681 13552 2@rpc/context.go:742  [n1] 846  closing [{n1: 127.0.0.1:41257 (default)} {n0: 127.0.0.1:41257 (default)}]
W220104 10:54:18.860924 12647 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 847  exiting heartbeat loop
    --- FAIL: TestTenantUpgradeFailure/upgrade_tenant_have_it_crash_then_resume (60.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantUpgradeFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go
+++ b/pkg/ccl/kvccl/kvtenantccl/tenant_upgrade_test.go
@@ -302,11 +302,14 @@ func TestTenantUpgradeFailure(t *testing.T) {
 			v2.String())
 		// Ensure that the tenant still works.
 		initialTenantRunner.CheckQueryResults(t, "SELECT * FROM t", [][]string{{"1"}, {"2"}})
+		// Use to wait for tenant crash leading to a clean up.
+		waitForTenantClose := make(chan struct{})
 		// Cause the upgrade to crash on v1.
 		go func() {
 			<-tenantStopperChannel
 			tenant.Close()
 			tenantInfo.v2onMigrationStopper.Stop(ctx)
+			waitForTenantClose <- struct{}{}
 		}()
 		// Upgrade the tenant cluster, but the upgrade
 		// will fail on v1.
@@ -314,6 +317,7 @@ func TestTenantUpgradeFailure(t *testing.T) {
 			".*(database is closed|failed to connect|closed network connection)+",
 			"SET CLUSTER SETTING version = $1",
 			clusterversion.TestingBinaryVersion.String())
+		<-waitForTenantClose
 		cleanup()
 		tenantInfo = mkTenant(t, initialTenantID, true /*existing*/)
 		tenant, cleanup = startAndConnectToTenant(t, tenantInfo)

